<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>qPCR Data Analysis • rtpcr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="qPCR Data Analysis">
<meta property="og:description" content="Various methods are employed for statistical analysis and graphical presentation of real-time PCR (quantitative PCR or qPCR) data. rtpcr handles amplification efficiency calculation, statistical analysis and graphical representation of real-time PCR data based on up to two reference genes. By accounting for amplification efficiency values, rtpcr was developed using a general calculation method described by Ganger et al. (2017) &lt;doi:10.1186/s12859-017-1949-5&gt; and Taylor et al. (2019) &lt;doi:10.1016/j.tibtech.2018.12.002&gt;, covering both the Livak and Pfaffl methods. Based on the experimental conditions, the functions of the rtpcr package use t-test (for experiments with a two-level factor), analysis of variance (ANOVA), analysis of covariance (ANCOVA) or analysis of repeated measure data to calculate the fold change (FC, Delta Delta Ct method) or relative expression (RE, Delta Ct method). The functions further provide standard errors and confidence intervals for means, apply statistical mean comparisons and present significance. To facilitate function application, different data sets were used as examples and the outputs were explained. ‘rtpcr’ package also provides bar plots using various controlling arguments. The rtpcr package is user-friendly and easy to work with and provides an applicable resource for analyzing real-time PCR data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">rtpcr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/vignette.html">rtpcr: a package for statistical analysis of real-time PCR data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mirzaghaderi/rtpcr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">


<div class="section level1">
<div class="page-header"><h1 id="section">
<a href="https://cran.r-project.org/web/packages/rtpcr/index.html" class="external-link"><img src="reference/figures/logo.jpg" align="right" height="120"></a><a class="anchor" aria-label="anchor" href="#section"></a>
</h1></div>
</div>
<div class="section level1">
<h1 id="rtpcr-a-package-for-statistical-analysis-and-graphical-presentation-of-real-time-pcr-data-in-r">rtpcr: a package for statistical analysis and graphical presentation of real-time PCR data in R<a class="anchor" aria-label="anchor" href="#rtpcr-a-package-for-statistical-analysis-and-graphical-presentation-of-real-time-pcr-data-in-r"></a>
</h1>
<div class="section level3">
<h3 id="author-ghader-mirzaghaderi">Author: Ghader Mirzaghaderi<a class="anchor" aria-label="anchor" href="#author-ghader-mirzaghaderi"></a>
</h3>
<p><em>A quick guide in PDF format to use rtpcr package:</em> <a href="man/figures/rtpcr-package.pdf"><img src="reference/figures/rtpcr.jpg"></a></p>
<p>Real-time polymerase chain reaction (real-time PCR), is widely used in biological research. Various analysis methods are employed on the real-time PCR data to measure the mRNA levels under different experimental conditions. ‘rtpcr’ package was developed for amplification efficiency calculation and statistical analysis of real-time PCR data in R. By accounting for up to two reference genes and amplification efficiency values, a general calculation methodology described by <a href="https://doi.org/10.1186/s12859-017-1949-5" class="external-link">Ganger et al. (2017)</a> and <a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">Taylor et al. 2019</a>, matching both <a href="https://doi.org/10.1006/meth.2001.1262" class="external-link">Livak and Schmittgen (2001)</a> and <a href="https://doi.org/10.1093/nar/30.9.e36" class="external-link">Pfaffl et al. (2002) </a> methods was used. Based on the experimental conditions, the functions of the ‘rtpcr’ package use a t-test (for experiments with a two-level factor) or analysis of variance (for cases where more than two levels or factors or a blocking factor exist) to calculate the fold change (FC) or relative expression (RE). The functions further provide standard deviations and confidence limits for means, apply statistical mean comparisons and present letter mean grouping. To facilitate function application, different data sets were used as examples and the outputs were explained. An outstanding feature of ‘rtpcr’ package is providing publication-ready bar plots with various controlling arguments for experiments with up to three different factors. The ‘rtpcr’ package was built based on a general method with various capabilities. It is user-friendly and easy to work with and provides an applicable resource for analyzing real-time PCR data in R. ‘rtpcr’ is a CRAN package although the development version of the package can be obtained through Github.</p>
</div>
</div>
<div class="section level1">
<h1 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h1>
<p>real-time PCR, also known as quantitative PCR (qPCR), is a powerful analytical tool that has revolutionized nucleic acid quantification. Among the various approaches developed for data analysis in real-time PCR, the Livak method (Livak and Schmittgen, 2001), also known as the <span class="math inline">2<sup>−<em>Δ</em><em>Δ</em><em>C</em><em>t</em></sup></span> method, stands out for its simplicity and widespread use. This method assumes that both the target and reference genes are amplified with efficiencies close to 100 %. On the other hand, the Pfaffl method (Pfaffl et al., 2002) offers a more flexible approach by accounting for differences in amplification efficiencies between the target and reference genes. This method adjusts the calculated expression ratio by incorporating the specific amplification efficiencies, thus providing a more accurate representation of the relative gene expression levels (Pfaffl et al., 2002). Both methods have their merits and limitations, and the choice between them depends on the experimental design and the precision required for the study. This paper aims to delve into the mathematical underpinnings of these methodologies, providing a comprehensive understanding of their applications and implications in real-time PCR analysis. Among the various approaches developed for data analysis in real-time PCR, the Livak method, also known as the <span class="math inline">2<sup>−<em>Δ</em><em>Δ</em><em>C</em><sub><em>t</em></sub></sup></span> method, stands out for its simplicity and widespread use.</p>
<p><span class="math display">Fold Change due to Treatment=</span></p>
<p><span class="math display">$$= \frac{2^{-(Ct_{target}-Ct_{ref})Tr}}{2^{-(Ct_{target}-Ct_{ref})Co}}$$</span></p>
<p><span class="math display"> = 2<sup>[−(<em>C</em><em>t</em><sub><em>t</em><em>a</em><em>r</em><em>g</em><em>e</em><em>t</em></sub> − <em>C</em><em>t</em><sub><em>r</em><em>e</em><em>f</em></sub>)<em>T</em><em>r</em> − (<em>C</em><em>t</em><sub><em>t</em><em>a</em><em>r</em><em>g</em><em>e</em><em>t</em></sub> − <em>C</em><em>t</em><sub><em>r</em><em>e</em><em>f</em></sub>)<em>C</em><em>o</em>]</sup></span></p>
<p><span class="math display"> = 2<sup>−(<em>Δ</em><em>C</em><em>t</em><sub><em>T</em><em>r</em></sub> − <em>Δ</em><em>C</em><em>t</em><sub><em>C</em><em>o</em></sub>)</sup></span></p>
<p><span class="math display"> = 2<sup>−<em>Δ</em><em>Δ</em><em>C</em><em>t</em></sup></span></p>
<p>where Tr is Treatment and Co is Control conditions, respectively. This method assumes that both the target and reference genes are amplified with efficiencies close to 100%, allowing for the relative quantification of gene expression levels (Livak and Schmittgen, 2001). On the other hand, the Pfaffl method offers a more flexible approach by accounting for differences in amplification efficiencies between the target and reference genes. This method adjusts the calculated expression ratio by incorporating the specific amplification efficiencies, thus providing a more accurate representation of the relative gene expression levels (Pfaffl et al., 2002).</p>
<p><span class="math display">$$\text{Fold Change due to Treatment} = \frac{E^{-(Ct_{Tr}-Ct_{Co})target}}{E^{-(Ct_{Tr}-Ct_{Co})ref}}$$</span></p>
</div>
<div class="section level1">
<h1 id="a-generalized-calculation-method">A generalized calculation method<a class="anchor" aria-label="anchor" href="#a-generalized-calculation-method"></a>
</h1>
<p>ΔC_t is the difference between the Ct value of the target gene and the Ct value of the referenec gene (e.g. ΔC_t = Cttarget−Ctref). The rtpcr package functions are mainly based on the calculation of efficiency-weighted ΔC_t (wΔC_t) values from target and reference gene Ct values which are weighted for their amplification efficiencies as below:</p>
<p><span class="math display"><em>w</em><em>Δ</em><em>C</em><em>t</em> = <em>l</em><em>o</em><em>g</em><sub>2</sub>(<em>E</em><sub><em>t</em><em>a</em><em>r</em><em>g</em><em>e</em><em>t</em></sub>).<em>C</em><em>t</em><sub><em>t</em><em>a</em><em>r</em><em>g</em><em>e</em><em>t</em></sub> − <em>l</em><em>o</em><em>g</em><sub>2</sub>(<em>E</em><sub><em>r</em><em>e</em><em>f</em></sub>).<em>C</em><em>t</em><sub><em>r</em><em>e</em><em>f</em></sub></span></p>
<p>From the mean wΔCt values over biological replicates, the expression of a target gene relative to the reference gene (e.g. calibrated by the reference gene) can be calculated for each condition as:</p>
<p><span class="math display">$$\text{Relative Expression (RE)} = 2^{-\overline{w\Delta Ct}}$$</span></p>
<p>Often, one condition is considered as calibrator condition. Examples are Treatment versus Control where Control is served as the calibrator, or time 0 versus time 1 (e.g. after 1 hour) and time 2 (e.g. after 2 hours) where time 0 is served as the reference or calibrator level. So, calibrator is the reference level or sample that all others are compared to. The fold change (FC) expression of a target gene for the reference or calibrator level is 1 because it is not changed compared to itself. The fold change expression of a target gene due to the treatment can be calculated as follows:</p>
<p><span class="math display">$$\text{Fold Change due to Treatment} = 2^{-(\overline{w\Delta Ct}{Tr}-\overline{w\Delta Ct}{Co})}$$</span></p>
<p>if wDCt values are calculated from the E values, these calculations match the formula of Pfaffl while if 2 (complete efficiency) is used instead the result matches the <span class="math inline">2<sup>−<em>Δ</em><em>Δ</em><em>C</em><em>t</em></sup></span> method. Standard error of the FC and RE means is calculated according to <a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">Taylor et al. 2019</a> in rtpcr package. Here, a brief methodology is presented but details about the wΔC_t calculations and statistical analysis are available in (Ganger et al., 2017). What is important here is that although the relative or fold change gene expression follows a lognormal distribution, a normal distribution is expected for the wΔCt or wΔΔCt values making it possible to apply t-test or analysis of variance to them. Following analysis, wΔCt values are statistically compared and standard deviations and confidence intervals are calculated, but the transformation y = 2^-x is applied in the final step to report the relative expression ratios, errors, and confidence limits.</p>
</div>
<div class="section level1">
<h1 id="installing-and-loading">Installing and loading<a class="anchor" aria-label="anchor" href="#installing-and-loading"></a>
</h1>
<p>The rtpcr package and source code are available for download from CRAN website (<a href="https://www.r-project.org" class="external-link uri">https://www.r-project.org</a>) under GPL-3 license. The <code>rtpcr</code> package can be installed by running the following code in R:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Installing from CRAN</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"rtpcr"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loading the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mirzaghaderi/rtpcr" class="external-link">rtpcr</a></span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, the latest version of the <code>rtpcr</code> package with the latest changes can be installed by running the following code in your R software. I strongly recommend to install the package with the vignette as it contains information about how to use the ‘rtpcr’ package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Installing from github</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mirzaghaderi/rtpcr"</span>, build_vignettes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If failed to build the vignette, just run the following code:</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mirzaghaderi/rtpcr"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Loading the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mirzaghaderi/rtpcr" class="external-link">rtpcr</a></span><span class="op">)</span></span></code></pre></div>
<p>Further information about how to use rtpcr package can be found <a href="https://cran.r-project.org/web/packages/rtpcr/vignettes/vignette.html" class="external-link">here </a>.</p>
</div>
<div class="section level1">
<h1 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h1>
<p>Email: gh.mirzaghaderi at uok.ac.ir</p>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<p>Livak Kenneth J, and Schmittgen TD. 2001. Analysis of Relative Gene Expression Data Using Real-Time Quantitative PCR and the Double Delta CT Method. Methods 25 (4). <a href="https://doi.org/10.1006/meth.2001.1262" class="external-link">doi.org/10.1006/meth.2001.1262</a>.</p>
<p>Ganger MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis of qPCR data and the application of simple blocking in qPCR experiments. BMC bioinformatics 18, 1-11. <a href="https://doi.org/10.1186/s12859-017-1949-5" class="external-link">doi.org/10.1186/s12859-017-1949-5</a>.</p>
<p>Pfaffl MW, Horgan GW, Dempfle L. 2002. Relative expression software tool (REST©) for group-wise comparison and statistical analysis of relative expression results in real-time PCR. Nucleic acids research 30, e36-e36. <a href="https://doi.org/10.1093/nar/30.9.e36" class="external-link">doi.org/10.1093/nar/30.9.e36</a>.</p>
<p>Taylor SC, Nadeau K, Abbasi M, Lachance C, Nguyen M, Fenrich, J. 2019. The ultimate qPCR experiment: producing publication quality, reproducible data the first time. Trends in Biotechnology, 37(7), 761-774<a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">doi.org/10.1016/j.tibtech.2018.12.002</a>.</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006. Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85). <a href="https://doi.org/10.1186/1471-2105-7-85" class="external-link">doi.org/10.1186/1471-2105-7-85</a>.</p>
<p>.</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=rtpcr" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/mirzaghaderi/rtpcr/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing rtpcr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ghader Mirzaghaderi <br><small class="roles"> Author, maintainer, copyright holder </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://cran.rstudio.com/web/packages/rtpcr/index.html" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/grand-total/rtpcr"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Ghader Mirzaghaderi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
