<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>rtpcr: a package for statistical analysis of real-time PCR data in R • rtpcr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="rtpcr: a package for statistical analysis of real-time PCR data in R">
<meta property="og:description" content="rtpcr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rtpcr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">rtpcr: a package for statistical analysis of real-time PCR data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mirzaghaderi/rtpcr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>rtpcr: a package for statistical analysis of
real-time PCR data in R</h1>
                        <h4 data-toc-skip class="author">Ghader
Mirzaghaderi</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mirzaghaderi/rtpcr/blob/HEAD/vignettes/vignette.Rmd" class="external-link"><code>vignettes/vignette.Rmd</code></a></small>
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Real-time polymerase chain reaction (real-time PCR) is widely used in
biological studies. Various analysis methods are employed on the
real-time PCR data to measure the mRNA levels under different
experimental conditions. ‘rtpcr’ package was developed for amplification
efficiency calculation, statistical analysis and bar plot representation
of real-time PCR data in R. By accounting for up to two reference genes
and amplification efficiency values, a general calculation methodology
described by <a href="https://doi.org/10.1186/s12859-017-1949-5" class="external-link">Ganger
et al. (2017)</a> and
<a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">Taylor et
al. (2019)</a>, matching both
<a href="https://doi.org/10.1006/meth.2001.1262" class="external-link">Livak and Schmittgen
(2001)</a> and <a href="https://doi.org/10.1093/nar/30.9.e36" class="external-link">Pfaffl et
al. (2002) </a> methods was used. Based on the experimental conditions,
the functions of the ‘rtpcr’ package use t-test (for experiments with a
two-level factor), analysis of variance, analysis of covariance (ANCOVA)
or analysis of repeated measure data to calculate the fold change (FC,
<span class="math inline">\({\Delta\Delta C_t}\)</span> method) or
relative expression (RE, <span class="math inline">\({\Delta
C_t}\)</span> method). The functions further provide standard errors and
confidence interval for means, apply statistical mean comparisons and
present significance. To facilitate function application, different data
sets were used as examples and the outputs were explained. An
outstanding feature of ‘rtpcr’ package is providing publication-ready
bar plots with various controlling arguments for experiments with up to
three different factors which are further editable by ggplot2
functions.</p>
</div>
<div class="section level2">
<h2 id="calculation-methods">Calculation methods<a class="anchor" aria-label="anchor" href="#calculation-methods"></a>
</h2>
<p>The basic method for expression estimation of a gene between
conditions relies on the calculation of fold differences by applying the
PCR amplification efficiency (E) and the threshold cycle (syn. crossing
point or Ct). Among the various approaches developed for data analysis
in real-time PCR, the Livak approach, also known as the <span class="math inline">\(2^{-\Delta\Delta C_t}\)</span> method, stands out
for its simplicity and widespread use where the fold change (FC)
exoression <span class="math inline">\((2^{-\Delta\Delta C_t})\)</span>
in Treatment (Tr) compared to Control (Co) condition is calculated
according to equation:</p>
<p><span class="math display">\[\begin{align*}
\text{Fold change} &amp; = 2^{-\Delta\Delta C_t} \\
&amp; = \frac{2^{-(C_{t_{\text{target}}}-C_{t_{\text{ref}}})_{Tr}}}
{2^{-(C_{t_{\text{target}}}-C_{t_{\text{ref}}})_{Co}}} \\
&amp; =2^{-[(C_{t_{\text{target}}}-C_{t_{\text{ref}}})_{\text{Tr}}-
{(C_{t_{\text{target}}}-C_{t_{\text{ref}}})}_{\text{Co}}]} \\
&amp; = 2^{-[{(\Delta C_t)_{Tr} - (\Delta C_t)_{Co}}]}
\end{align*}\]</span></p>
<p>Here, <span class="math inline">\(\Delta C_t\)</span> is the
difference between target Ct and reference Ct values for a given sample.
Livak method assumes that both the target and reference genes are
amplified with efficiencies close to 100%, allowing for the relative
quantification of gene expression levels.</p>
<p>On the other hand, the Pfaffl method offers a more flexible approach
by accounting for differences in amplification efficiencies between the
target and reference genes. This method adjusts the calculated
expression ratio by incorporating the specific amplification
efficiencies, thus providing a more accurate representation of the
relative gene expression levels.</p>
<p><span class="math display">\[\text{Fold change} =
\frac{E^{-(C_{t_{\text{Tr}}}-C_{t_{\text{Co}}})_{target}}}
{E^{-(C_{t_{\text{Tr}}}-C_{t_{\text{Co}}})_{ref}}}\]</span></p>
</div>
<div class="section level2">
<h2 id="a-generalized-calculation-method">A generalized calculation method<a class="anchor" aria-label="anchor" href="#a-generalized-calculation-method"></a>
</h2>
<p>The <code>rtpcr</code> package was developed for the R environment in
the major operating systems. The package functions are mainly based on
the calculation of efficiency-weighted <span class="math inline">\(\Delta C_t\)</span> <span class="math inline">\((w\Delta C_t)\)</span> values from target and
reference gene Ct (equation 3). <span class="math inline">\(w\Delta
C_t\)</span> values are weighted for the amplification efficiencies as
described by Ganger et al. (2017) except that log2 is used instead of
log10:</p>
<p><span class="math display">\[w\Delta Ct
=\log_{2}(E_{target}).Ct_{target}-\log_{2}(E_{ref}).Ct_{ref}\]</span></p>
<p>The relative expression of the target gene normalized to that of
reference gene(s) within the same sample or condition is called relative
expression (RE). From the mean <span class="math inline">\(w\Delta
C_t\)</span> values over biological replicates, RE of a target gene can
be calculated for each condition according to the equation</p>
<p><span class="math display">\[\text{Relative Expression} =
2^{-\overline{w\Delta Ct}}\]</span> Relative expression is only
calibrated for the reference gene(s) and not for a control condition.
However, often one condition is considered as calibrator and the fold
change (FC) expression in other conditions is calculated relative to the
calibrator. Examples are Treatment versus Control where Control is
served as the calibrator, or time 0 versus time 1 (e.g. after 1 hour)
and time 2 (e.g. after 2 hours) where time 0 is served as the reference
or calibrator level. So, calibrator is the reference level or sample
that all others are compared to. The fold change (FC) expression of a
target gene for the reference or calibrator level is 1 because it is not
changed compared to itself. The fold change expression of a target gene
due to the treatment can be calculated as follows:</p>
<p><span class="math display">\[\text{Fold Change due to
Treatment}=2^{-(\overline{w\Delta Ct}_{\text{Tr}}-{\overline{w\Delta
Ct}_{\text{Co}}})}\]</span></p>
<p><code>qpcrTTEST</code> and <code>qpcrTTESTplot</code> functions
calculate FC for multi-genes-two conditional cases,
<code>qpcrANOVAFC</code> represents FC for single-gene-factorial
(single- or multi-factor) experiments, and <code>qpcrREPEATED</code>
calculates FC for the repeated measure data. If <span class="math inline">\(w \Delta C_t\)</span> values is calculated from
the E values, the calculations match the formula of Pfaffl while if 2
(complete efficiency) be used instead, the result match the <span class="math inline">\(2^{-\Delta\Delta C_t}\)</span> method. In any case
we called these as Fold Change in the outputs of <code>rtpcr</code>.
Under factorial experiments where the calculation of the expression of
the target gene relative to the reference gene (called Relative
Expression) in each condition is desired, <code>qpcrANOVARE</code>,
<code>oneFACTORplot</code>, <code>twoFACTORplot</code> and
<code>threeFACTORplot</code> functions were developed for ANOVA
analysis, and representing the plots from single, double or triple
factor experiments, respectively. The last three functions generate
<code>ggplot2</code>-derived graphs based on the output of the
<code>qpcrANOVARE</code> function. If available, the blocking factor can
also be handled by <code>qpcrANOVARE</code>, <code>qpcrANOVAFC</code>
and <code>qpcrREPEATED</code> functions. Standard error of the FC and RE
means is calculated according to
<a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">Taylor et
al. (2019)</a> in <code>rtpcr</code> package. Here, a brief methodology
is presented but details about the <span class="math inline">\(w\Delta
C_t\)</span> calculations and statistical analysis are available in
<a href="https://doi.org/10.1186/s12859-017-1949-5" class="external-link">Ganger et
al. (2017)</a>. Importantly, because both the RE or FC gene expression
values follow a lognormal distribution, a normal distribution is
expected for the <span class="math inline">\(w \Delta C_t\)</span>
values making it possible to apply t-test or analysis of variance to
them. Following analysis, <span class="math inline">\(w\Delta
C_t\)</span> values are statistically compared and standard deviations
and confidence interval are calculated, but the transformation <span class="math inline">\(y = 2^{-x}\)</span> is applied in the final step
in order to report the results.</p>
</div>
<div class="section level2">
<h2 id="installing-and-loading">Installing and loading<a class="anchor" aria-label="anchor" href="#installing-and-loading"></a>
</h2>
<p>The <code>rtpcr</code> package can be installed and loaded using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"rtpcr"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mirzaghaderi/rtpcr" class="external-link">rtpcr</a></span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, the <code>rtpcr</code> with the latest changes can be
installed by running the following code in your R software:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install `rtpcr` from github (under development)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mirzaghaderi/rtpcr"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># I strongly recommend to install the package with the vignette as it contains information about how to use the 'rtpcr' package. Through the following code, Vignette is installed as well.</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mirzaghaderi/rtpcr"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-structure-and-column-arrangement">Data structure and column arrangement<a class="anchor" aria-label="anchor" href="#data-structure-and-column-arrangement"></a>
</h2>
<p>To use the functions, input data should be prepared in the right
format with appropriate column arrangement. The correct column
arrangement is shown in Table 1 and Table 2. For
<code>qpcrANOVAFC</code> or <code>qpcrANOVARE</code> analysis, ensure
that each line in the data set belongs to a separate individual or
biological replicate reflecting a non-repeated measure experiment.</p>
<p><em>Table 1. Data structure and column arrangement required for
‘rtpcr’ package. rep: technical replicate; targetE and refE:
amplification efficiency columns for target and reference genes,
respectively. targetCt and refCt: target gene and reference gene Ct
columns, respectively. factors (up to three factors is allowed):
experimental factors.</em></p>
<table class="table">
<colgroup>
<col width="20%">
<col width="38%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th align="left">Experiment type</th>
<th align="left">Column arrangement of the input data</th>
<th align="left">Example in the package</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Amplification efficiency</td>
<td align="left">Dilutions - geneCt …</td>
<td align="left">data_efficiency</td>
</tr>
<tr class="even">
<td align="left">t-test (accepts multiple genes)</td>
<td align="left">condition (put the control level first) - gene (put
reference gene(s) last.)- efficiency - Ct</td>
<td align="left">data_ttest</td>
</tr>
<tr class="odd">
<td align="left">Factorial (Up to three factors)</td>
<td align="left">factor1 - rep - targetE - targetCt - refE - refCt</td>
<td align="left">data_1factor</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">factor1 - factor2 - rep - targetE - targetCt - refE -
refCt</td>
<td align="left">data_2factor</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">factor1 - factor2 - factor3 - rep - targetE - targetCt
- refE - refCt</td>
<td align="left">data_3factor</td>
</tr>
<tr class="even">
<td align="left">Factorial with blocking</td>
<td align="left">factor1 - block - rep - targetE - targetCt - refE -
refCt</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">factor1 - factor2 - block - rep - targetE - targetCt -
refE - refCt</td>
<td align="left">data_2factorBlock</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">factor1 - factor2 - factor3 - block - rep - targetE -
targetCt - refE - refCt</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Two reference genes</td>
<td align="left">. . . . . . rep - targetE - targetCt - ref1E - ref1Ct -
ref2E - ref2Ct</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">calculating biological replicated</td>
<td align="left">. . . . . . biologicalRep - techcicalRep - Etarget -
targetCt - Eref - refCt</td>
<td align="left">data_withTechRep</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">. . . . . . biologicalRep - techcicalRep - Etarget -
targetCt - ref1E - ref1Ct - ref2E - ref2Ct</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>NOTE: For <code>qpcrANOVAFC</code> or <code>qpcrANOVARE</code>
analysis, each line in the input data set belongs to a separate
individual or biological replicate reflecting a non-repeated measure
experiment.</p>
<p><em>Table 2. Repeated measure data structure and column arrangement
required for the <code>qpcrREPEATED</code> function. targetE and refE:
amplification efficiency columns for target and reference genes,
respectively. targetCt and refCt: Ct columns for target and reference
genes, respectively. In the “id” column, a unique number is assigned to
each individual, e.g. all the three number 1 indicate a single
individual.</em></p>
<table class="table">
<colgroup>
<col width="20%">
<col width="38%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th align="left">Experiment type</th>
<th align="left">Column arrangement of the input data</th>
<th align="left">Example in the package</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Repeated measure</td>
<td align="left">id - time - targetE - targetCt - ref1E - ref1Ct</td>
<td align="left">data_repeated_measure_1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">id - time - targetE - targetCt - ref1E - ref1Ct - ref2E
- ref2Ct</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Repeated measure</td>
<td align="left">id - treatment - time - targetE - targetCt - ref1E -
ref1Ct</td>
<td align="left">data_repeated_measure_2</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">id - treatment - time - targetE - targetCt - ref1E -
ref1Ct - ref2E - ref2Ct</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>To see list of data in the <code>rtpcr</code> package run
<code>data(package = "rtpcr")</code>. Example data sets can be presented
by running the name of each data set. A description of the columns names
in each data set is called by “?” followed by the names of the data set,
for example <code><a href="../reference/data_1factor.html">?data_1factor</a></code></p>
</div>
<div class="section level2">
<h2 id="functions-usage">functions usage<a class="anchor" aria-label="anchor" href="#functions-usage"></a>
</h2>
<p>To simplify <code>rtpcr</code> usage, examples for using the
functions are presented below.</p>
<p><em>Table 3. Functions and examples for using them.</em></p>
<table class="table">
<colgroup>
<col width="23%">
<col width="38%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">function</th>
<th align="left">Analysis</th>
<th align="left">Example (see package help for more arguments)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">efficiency</td>
<td align="left">Efficiency, standard curves and related statistics</td>
<td align="left">efficiency(data_efficiency)</td>
</tr>
<tr class="even">
<td align="left">meanTech</td>
<td align="left">Calculating the mean of technical replicates</td>
<td align="left">meanTech(data_withTechRep, groups = 1:4)</td>
</tr>
<tr class="odd">
<td align="left">qpcrANOVAFC</td>
<td align="left">FC and bar plot of the target gene (one or
multi-factorial experiments)</td>
<td align="left">qpcrANOVAFC(data_1factor, numberOfrefGenes = 1,
mainFactor.column = 1, mainFactor.level.order = c(“L1”, “L2”, “L3”)</td>
</tr>
<tr class="even">
<td align="left">oneFACTORplot</td>
<td align="left">Bar plot of the relative gene expression from a
one-factor experiment</td>
<td align="left">out &lt;- qpcrANOVARE(data_1factor, numberOfrefGenes =
1)$Result; oneFACTORplot(out, errorbar = “se”)</td>
</tr>
<tr class="odd">
<td align="left">qpcrANOVARE</td>
<td align="left">Analysis of Variance of the qpcr data</td>
<td align="left">qpcrANOVARE(data_3factor, numberOfrefGenes = 1)</td>
</tr>
<tr class="even">
<td align="left">qpcrTTEST</td>
<td align="left">Computing the fold change and related statistics</td>
<td align="left">qpcrTTEST(data_ttest, numberOfrefGenes = 1, paired =
FALSE, var.equal = TRUE)</td>
</tr>
<tr class="odd">
<td align="left">qpcrTTESTplot</td>
<td align="left">Bar plot of the average fold change of the target
genes</td>
<td align="left">qpcrTTESTplot(data_ttest, numberOfrefGenes = 1, order =
c(“C2H2-01”, “C2H2-12”, “C2H2-26”))</td>
</tr>
<tr class="even">
<td align="left">threeFACTORplot</td>
<td align="left">Bar plot of the relative gene expression from a
three-factor experiment</td>
<td align="left">res &lt;- qpcrANOVARE(data_3factor, numberOfrefGenes =
1)$Result; threeFACTORplot(res,</td>
</tr>
<tr class="odd">
<td align="left">arrangement = c(3, 1, 2), errorbar = “se”)</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">twoFACTORplot</td>
<td align="left">Bar plot of the relative gene expression from a
two-factor experiment</td>
<td align="left">res &lt;- qpcrANOVARE(data_2factor, numberOfrefGenes =
1)$Result; twoFACTORplot(res, x.axis.factor = Genotype, group.factor =
Drought, errorbar = “se”)</td>
</tr>
<tr class="odd">
<td align="left">qpcrREPEATED</td>
<td align="left">Bar plot of the fold change expression for repeated
measure observations (taken over time from each individual)</td>
<td align="left">qpcrREPEATED(data_repeated_measure_2, numberOfrefGenes
= 1), factor = “time”</td>
</tr>
</tbody>
</table>
<p><em>see package help for more arguments including the number of
reference genes, levels arrangement, blocking, and arguments for
adjusting the bar plots.</em></p>
</div>
<div class="section level2">
<h2 id="amplification-efficiency-data-analysis">Amplification efficiency data analysis<a class="anchor" aria-label="anchor" href="#amplification-efficiency-data-analysis"></a>
</h2>
<div class="section level3">
<h3 id="sample-data-of-amplification-efficiency">Sample data of amplification efficiency<a class="anchor" aria-label="anchor" href="#sample-data-of-amplification-efficiency"></a>
</h3>
<p>To calculate the amplification efficiencies of a target and a
reference gene, a data frame should be prepared with 3 columns of
dilutions, target gene Ct values, and reference gene Ct values,
respectively, as shown below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_efficiency</span></span></code></pre></div>
<pre><code><span><span class="co">##    dilutions  C2H2.26 C2H2.01    GAPDH</span></span>
<span><span class="co">## 1       1.00 25.57823   24.25 22.60794</span></span>
<span><span class="co">## 2       1.00 25.53636   24.13 22.68348</span></span>
<span><span class="co">## 3       1.00 25.50280   24.04 22.62602</span></span>
<span><span class="co">## 4       0.50 26.70615   25.56 23.67162</span></span>
<span><span class="co">## 5       0.50 26.72720   25.43 23.64855</span></span>
<span><span class="co">## 6       0.50 26.86921   26.01 23.70494</span></span>
<span><span class="co">## 7       0.20 28.16874   27.37 25.11064</span></span>
<span><span class="co">## 8       0.20 28.06759   26.94 25.11985</span></span>
<span><span class="co">## 9       0.20 28.10531   27.14 25.10976</span></span>
<span><span class="co">## 10      0.10 29.19743   28.05 26.16919</span></span>
<span><span class="co">## 11      0.10 29.49406   28.89 26.15119</span></span>
<span><span class="co">## 12      0.10 29.07117   28.32 26.15019</span></span>
<span><span class="co">## 13      0.05 30.16878   29.50 27.11533</span></span>
<span><span class="co">## 14      0.05 30.14193   29.93 27.13934</span></span>
<span><span class="co">## 15      0.05 30.11671   29.71 27.16338</span></span>
<span><span class="co">## 16      0.02 31.34969   30.69 28.52016</span></span>
<span><span class="co">## 17      0.02 31.35254   30.54 28.57228</span></span>
<span><span class="co">## 18      0.02 31.34804   30.04 28.53100</span></span>
<span><span class="co">## 19      0.01 32.55013   31.12 29.49048</span></span>
<span><span class="co">## 20      0.01 32.45329   31.29 29.48433</span></span>
<span><span class="co">## 21      0.01 32.27515   31.15 29.26234</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="calculating-amplification-efficiency">Calculating amplification efficiency<a class="anchor" aria-label="anchor" href="#calculating-amplification-efficiency"></a>
</h3>
<p>Amplification efficiency in PCR can be either defined as percentage
(from 0 to 1) or as time of PCR product increase per cycle (from 1 to
2). in the <code>rtpcr</code> package, the amplification efficiency (E)
has been referred to times of PCR product increase (1 to 2). A complete
efficiency is equal to 2. If dilutions and Ct values are available for a
number of genes, the <code>efficiency</code> function calculates the
amplification efficiency of genes and presents the related standard
curves along with the Slope, Efficiency, and R2 statistics. The function
also does pairwise comparisons of the slopes for the genes. For this, a
regression line is fitted forst using the <span class="math inline">\(\Delta C_t\)</span> values of each pair of
genes.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/efficiency.html">efficiency</a></span><span class="op">(</span><span class="va">data_efficiency</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $Efficiency</span></span>
<span><span class="co">##      Gene     Slope        R2        E</span></span>
<span><span class="co">## 1 C2H2.26 -3.388094 0.9965504 1.973110</span></span>
<span><span class="co">## 2 C2H2.01 -3.528125 0.9713914 1.920599</span></span>
<span><span class="co">## 3   GAPDH -3.414551 0.9990278 1.962747</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Slope_compare</span></span>
<span><span class="co">## $emtrends</span></span>
<span><span class="co">##  variable log10(dilutions).trend     SE df lower.CL upper.CL</span></span>
<span><span class="co">##  C2H2.26                   -3.39 0.0856 57    -3.56    -3.22</span></span>
<span><span class="co">##  C2H2.01                   -3.53 0.0856 57    -3.70    -3.36</span></span>
<span><span class="co">##  GAPDH                     -3.41 0.0856 57    -3.59    -3.24</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Confidence level used: 0.95 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $contrasts</span></span>
<span><span class="co">##  contrast          estimate    SE df t.ratio p.value</span></span>
<span><span class="co">##  C2H2.26 - C2H2.01   0.1400 0.121 57   1.157  0.4837</span></span>
<span><span class="co">##  C2H2.26 - GAPDH     0.0265 0.121 57   0.219  0.9740</span></span>
<span><span class="co">##  C2H2.01 - GAPDH    -0.1136 0.121 57  -0.938  0.6186</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## P value adjustment: tukey method for comparing a family of 3 estimates </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $plot</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-4-1.png" alt="Standard curve and the amplification efficiency analysis of genes. Required iput data include dilutions and Ct value columns for different genes." width="480"><p class="caption">
Standard curve and the amplification efficiency analysis of genes.
Required iput data include dilutions and Ct value columns for different
genes.
</p>
</div>
<p>Note: It is advised that the amplification efficiency be calculated
for each cDNA sample because the amplification efficiency not only
depends on the PCR mic and primer characteristics, but also varies among
different cDNA samples.</p>
</div>
</div>
<div class="section level2">
<h2 id="expression-data-analysis">Expression data analysis<a class="anchor" aria-label="anchor" href="#expression-data-analysis"></a>
</h2>
<div class="section level3">
<h3 id="target-genes-in-two-conditions-t-test">Target genes in two conditions (t-test)<a class="anchor" aria-label="anchor" href="#target-genes-in-two-conditions-t-test"></a>
</h3>
<div class="section level4">
<h4 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h4>
<p>When a target gene is assessed under two different conditions (for
example Control and treatment), it is possible to calculate the average
fold change expression <span class="math inline">\(({\Delta \Delta
C_t}\)</span> method) of the target gene in treatment relative to
control conditions. For this, the data should be prepared according to
the following data set consisting of 4 columns belonging to condition
levels, E (efficiency), genes and Ct values, respectively. Each Ct value
is the mean of technical replicates. Complete amplification efficiencies
of 2 have been assumed here for all wells but the calculated
efficiencies can be used instead.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_ttest</span></span></code></pre></div>
<pre><code><span><span class="co">##    Condition    Gene E    Ct</span></span>
<span><span class="co">## 1    control C2H2-26 2 31.26</span></span>
<span><span class="co">## 2    control C2H2-26 2 31.01</span></span>
<span><span class="co">## 3    control C2H2-26 2 30.97</span></span>
<span><span class="co">## 4  treatment C2H2-26 2 32.65</span></span>
<span><span class="co">## 5  treatment C2H2-26 2 32.03</span></span>
<span><span class="co">## 6  treatment C2H2-26 2 32.40</span></span>
<span><span class="co">## 7    control C2H2-01 2 31.06</span></span>
<span><span class="co">## 8    control C2H2-01 2 30.41</span></span>
<span><span class="co">## 9    control C2H2-01 2 30.97</span></span>
<span><span class="co">## 10 treatment C2H2-01 2 28.85</span></span>
<span><span class="co">## 11 treatment C2H2-01 2 28.93</span></span>
<span><span class="co">## 12 treatment C2H2-01 2 28.90</span></span>
<span><span class="co">## 13   control C2H2-12 2 28.50</span></span>
<span><span class="co">## 14   control C2H2-12 2 28.40</span></span>
<span><span class="co">## 15   control C2H2-12 2 28.80</span></span>
<span><span class="co">## 16 treatment C2H2-12 2 27.90</span></span>
<span><span class="co">## 17 treatment C2H2-12 2 28.00</span></span>
<span><span class="co">## 18 treatment C2H2-12 2 27.90</span></span>
<span><span class="co">## 19   control     ref 2 28.87</span></span>
<span><span class="co">## 20   control     ref 2 28.42</span></span>
<span><span class="co">## 21   control     ref 2 28.53</span></span>
<span><span class="co">## 22 treatment     ref 2 28.31</span></span>
<span><span class="co">## 23 treatment     ref 2 29.14</span></span>
<span><span class="co">## 24 treatment     ref 2 28.63</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="data-analysis-under-two-conditions">Data analysis under two conditions<a class="anchor" aria-label="anchor" href="#data-analysis-under-two-conditions"></a>
</h4>
<p>Here, the above data set was used for the Fold Change expression
analysis of the target genes using the <code>qpcrTTEST</code> function.
This function performs a t-test-based analysis of any number of genes
that have been evaluated under control and treatment conditions. The
output is a table of target gene names, fold changes confidence limits,
and the t.test derived p-values. The <code>qpcrTTEST</code> function
includes the <code>var.equal</code> argument. When set to
<code>FALSE</code>, <code>t.test</code> is performed under the unequal
variances hypothesis. Furthermore, the samples in qPCR may be unpaired
or paired so the analysis can be done for unpaired or paired conditions.
Paired samples refer to a situation where the measurements are made on
the same set of subjects or individuals. This could occur if the data is
acquired from the same set of individuals before and after a treatment.
In such cases, the paired t-test is used for statistical comparisons by
setting the t-test <code>paired</code> argument to
<code>TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/qpcrTTEST.html">qpcrTTEST</a></span><span class="op">(</span><span class="va">data_ttest</span>, </span>
<span>          numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          paired <span class="op">=</span> <span class="cn">F</span>, </span>
<span>          var.equal <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $Raw_data</span></span>
<span><span class="co">##    Var2  wDCt</span></span>
<span><span class="co">## 1     1  2.39</span></span>
<span><span class="co">## 2     1  2.59</span></span>
<span><span class="co">## 3     1  2.44</span></span>
<span><span class="co">## 4     1  4.34</span></span>
<span><span class="co">## 5     1  2.89</span></span>
<span><span class="co">## 6     1  3.77</span></span>
<span><span class="co">## 7     2  2.19</span></span>
<span><span class="co">## 8     2  1.99</span></span>
<span><span class="co">## 9     2  2.44</span></span>
<span><span class="co">## 10    2  0.54</span></span>
<span><span class="co">## 11    2 -0.21</span></span>
<span><span class="co">## 12    2  0.27</span></span>
<span><span class="co">## 13    3 -0.37</span></span>
<span><span class="co">## 14    3 -0.02</span></span>
<span><span class="co">## 15    3  0.27</span></span>
<span><span class="co">## 16    3 -0.41</span></span>
<span><span class="co">## 17    3 -1.14</span></span>
<span><span class="co">## 18    3 -0.73</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Result</span></span>
<span><span class="co">##      Gene     FC    LCL    UCL pvalue     se Lower.se Upper.se</span></span>
<span><span class="co">## 1 C2H2-26 0.4373 0.1926 0.9927 0.0488 0.4218   0.3264   0.5858</span></span>
<span><span class="co">## 2 C2H2-01 4.0185 2.4598 6.5649 0.0014 0.2193   3.4518   4.6782</span></span>
<span><span class="co">## 3 C2H2-12 1.6472 0.9595 2.8279 0.0624 0.2113   1.4228   1.9070</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="generating-plot">Generating plot<a class="anchor" aria-label="anchor" href="#generating-plot"></a>
</h4>
<p>The <code>qpcrTTESTplot</code> function generates a bar plot of fold
changes (FC) and confidence intervals for the target genes. the
<code>qpcrTTESTplot</code> function accepts any number of genes and any
replicates. The <code>qpcrTTESTplot</code> function automatically puts
appropriate signs of **, * on top of the plot columns based on the
output p-values.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Producing the plot</span></span>
<span><span class="va">t1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrTTESTplot.html">qpcrTTESTplot</a></span><span class="op">(</span><span class="va">data_ttest</span>,</span>
<span>              numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              fontsizePvalue <span class="op">=</span> <span class="fl">4</span>,</span>
<span>              errorbar <span class="op">=</span> <span class="st">"ci"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Producing the plot: specifying gene order</span></span>
<span><span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrTTESTplot.html">qpcrTTESTplot</a></span><span class="op">(</span><span class="va">data_ttest</span>,</span>
<span>              numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C2H2-01"</span>, <span class="st">"C2H2-12"</span>, <span class="st">"C2H2-26"</span><span class="op">)</span>,</span>
<span>              paired <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              var.equal <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>              fill <span class="op">=</span> <span class="st">"palegreen"</span>,</span>
<span>              y.axis.adjust <span class="op">=</span> <span class="fl">3</span>,</span>
<span>              y.axis.by <span class="op">=</span> <span class="fl">2</span>,</span>
<span>              ylab <span class="op">=</span> <span class="st">"Average Fold Change"</span>,</span>
<span>              xlab <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>              fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/multiplot.html">multiplot</a></span><span class="op">(</span><span class="va">t1</span>, <span class="va">t2</span>, cols <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $plot</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $plot</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"A"</span>, x <span class="op">=</span> <span class="fl">0.02</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"B"</span>, x <span class="op">=</span> <span class="fl">0.52</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-7-1.png" alt="Average Fold changes of three target genes relative to the control condition computed by unpaired t-tests via `qpcrTTESTplot` function. Confidence interval (ci) and standard error (se) has been used as error bar in 'A' and 'B', respectively." width="768"><p class="caption">
Average Fold changes of three target genes relative to the control
condition computed by unpaired t-tests via <code>qpcrTTESTplot</code>
function. Confidence interval (ci) and standard error (se) has been used
as error bar in ‘A’ and ‘B’, respectively.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="uni--or-multi-factorial-experiments-fold-change-fc-analysis">Uni- or multi-factorial experiments: Fold Change (FC) analysis<a class="anchor" aria-label="anchor" href="#uni--or-multi-factorial-experiments-fold-change-fc-analysis"></a>
</h3>
<p>In the rtpcr package, the fold change analysis of a target gene is
applied based on analysis of variance; ANOVA (or analysis of covariance;
ANCOVA) using the <code>qpcrANOVAFC</code> function. the default
argument is <code>anova</code> but can be changes to <code>ancova</code>
as well. the ancova analysis is suitable when the levels of a factor are
also affected by an uncontrolled quantitative covariate. For example,
suppose that wDCt of a target gene in a plant is affected by
temperature. The gene may also be affected by drought. since we already
know that temperature affects the target gene, we are interesting now if
the gene expression is also altered by the drought levels. We can design
an experiment to understand the gene behavior at both temperature and
drought levels at the same time. The drought is another factor (the
covariate) that may affect the expression of our gene under the levels
of the first factor i.e. temperature. The data of such an experiment can
be analyzed by both ANCOVA or factorial ANOVA using
<code>qpcrANOVAFC</code> function. The function also works for one
factor experiment as well. Bar plot of fold changes (FC values) along
with the 95% confidence interval is also returned by the
<code>qpcrANOVAFC</code> function. There is also a function called
<code>oneFACTORplot</code> which returns FC values and related plot for
a one-factor-experiment with more than two levels.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See sample data</span></span>
<span><span class="va">data_2factor</span></span></code></pre></div>
<pre><code><span><span class="co">##    Genotype Drought Rep EPO  POCt EGAPDH GAPDHCt</span></span>
<span><span class="co">## 1         R      D0   1   2 33.30      2   31.53</span></span>
<span><span class="co">## 2         R      D0   2   2 33.39      2   31.57</span></span>
<span><span class="co">## 3         R      D0   3   2 33.34      2   31.50</span></span>
<span><span class="co">## 4         R      D1   1   2 32.73      2   31.30</span></span>
<span><span class="co">## 5         R      D1   2   2 32.46      2   32.55</span></span>
<span><span class="co">## 6         R      D1   3   2 32.60      2   31.92</span></span>
<span><span class="co">## 7         R      D2   1   2 33.48      2   33.30</span></span>
<span><span class="co">## 8         R      D2   2   2 33.27      2   33.37</span></span>
<span><span class="co">## 9         R      D2   3   2 33.32      2   33.35</span></span>
<span><span class="co">## 10        S      D0   1   2 26.85      2   26.94</span></span>
<span><span class="co">## 11        S      D0   2   2 28.17      2   27.69</span></span>
<span><span class="co">## 12        S      D0   3   2 27.99      2   27.39</span></span>
<span><span class="co">## 13        S      D1   1   2 30.41      2   28.70</span></span>
<span><span class="co">## 14        S      D1   2   2 29.49      2   28.66</span></span>
<span><span class="co">## 15        S      D1   3   2 29.98      2   28.71</span></span>
<span><span class="co">## 16        S      D2   1   2 29.03      2   30.61</span></span>
<span><span class="co">## 17        S      D2   2   2 28.73      2   30.20</span></span>
<span><span class="co">## 18        S      D2   3   2 28.83      2   30.49</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/qpcrANOVAFC.html">qpcrANOVAFC</a></span><span class="op">(</span><span class="va">data_2factor</span>, </span>
<span>           numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, </span>
<span>           block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           analysisType <span class="op">=</span> <span class="st">"ancova"</span>,</span>
<span>           mainFactor.column <span class="op">=</span> <span class="fl">2</span>,</span>
<span>           fontsizePvalue <span class="op">=</span> <span class="fl">4</span>,</span>
<span>           x.axis.labels.rename <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrANOVAFC(data_2factor, numberOfrefGenes = 1, block = NULL, : The</span></span>
<span><span class="co">## D0 level was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## ANOVA table </span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##                   Sum Sq Mean Sq NumDF  DenDF F value    Pr(&gt;F)    </span></span>
<span><span class="co">## Drought          12.9530  6.4765     2 9.9998  41.691 1.408e-05 ***</span></span>
<span><span class="co">## Genotype          3.0505  3.0505     1 9.9998  19.637  0.001272 ** </span></span>
<span><span class="co">## Drought:Genotype  4.5454  2.2727     2 9.9998  14.630  0.001072 ** </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ANCOVA table</span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##           Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span></span>
<span><span class="co">## Genotype  3.0505  3.0505     1    14   6.649 0.0218703 *  </span></span>
<span><span class="co">## Drought  12.9530  6.4765     2    14  14.117 0.0004398 ***</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##   contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1       D0 1.0000 1.0000     0.0000 0.0000 0.3445   0.7876   1.2697</span></span>
<span><span class="co">## 2 D1 vs D0 1.0705 0.8051     0.5266 2.1763 0.2631   0.8920   1.2847</span></span>
<span><span class="co">## 3 D2 vs D0 3.5967 0.0003 *** 1.7693 7.3116 0.3576   2.8071   4.6084</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change plot of the main factor levels</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-8-1.png" alt="Statistical table and figure of the Fold change expression of a gene in three different levels of Drough stress relative to the D0 as reference or calibrator level produced by the `qpcrANOVAFC` function. The other factor i.e. Genotype has been concidered as covariate." width="480"><p class="caption">
Statistical table and figure of the Fold change expression of a gene in
three different levels of Drough stress relative to the D0 as reference
or calibrator level produced by the <code>qpcrANOVAFC</code> function.
The other factor i.e. Genotype has been concidered as covariate.
</p>
</div>
</div>
<div class="section level3">
<h3 id="uni--or-multi-factorial-experiments-relative-expression-re-analysis">Uni- or multi-factorial experiments: Relative Expression (RE)
analysis<a class="anchor" aria-label="anchor" href="#uni--or-multi-factorial-experiments-relative-expression-re-analysis"></a>
</h3>
<p>the <code>qpcrANOVARE</code> function performs Relative Expression
(RE) analysis for uni- or multi-factorial experiments in which all
factor level combinations are used as treatments. The input data set
should be prepared as shown in table 1. Factor columns should be
presented first followed by blocking factor (if available), biological
replicates and efficiency and Ct values of target and reference gene(s).
The example data set below (<code>data_3factor</code>) represents
amplification efficiency and Ct values for target and reference genes
under three grouping factors (two different cultivars, three drought
levels, and the presence or absence of bacteria). Here, the efficiency
value of 2 has been used for all wells, but the calculated efficiencies
can be used instead.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See a sample dataset</span></span>
<span><span class="va">data_3factor</span></span></code></pre></div>
<pre><code><span><span class="co">##    Type Conc SA Replicate EPO  POCt EGAPDH GAPDHCt</span></span>
<span><span class="co">## 1     R    L A1         1   2 33.30      2   31.53</span></span>
<span><span class="co">## 2     R    L A1         2   2 33.39      2   31.57</span></span>
<span><span class="co">## 3     R    L A1         3   2 33.34      2   31.50</span></span>
<span><span class="co">## 4     R    L A2         1   2 34.01      2   31.48</span></span>
<span><span class="co">## 5     R    L A2         2   2 36.82      2   31.44</span></span>
<span><span class="co">## 6     R    L A2         3   2 35.44      2   31.46</span></span>
<span><span class="co">## 7     R    M A1         1   2 32.73      2   31.30</span></span>
<span><span class="co">## 8     R    M A1         2   2 32.46      2   32.55</span></span>
<span><span class="co">## 9     R    M A1         3   2 32.60      2   31.92</span></span>
<span><span class="co">## 10    R    M A2         1   2 33.37      2   31.19</span></span>
<span><span class="co">## 11    R    M A2         2   2 33.12      2   31.94</span></span>
<span><span class="co">## 12    R    M A2         3   2 33.21      2   31.57</span></span>
<span><span class="co">## 13    R    H A1         1   2 33.48      2   33.30</span></span>
<span><span class="co">## 14    R    H A1         2   2 33.27      2   33.37</span></span>
<span><span class="co">## 15    R    H A1         3   2 33.32      2   33.35</span></span>
<span><span class="co">## 16    R    H A2         1   2 32.53      2   33.47</span></span>
<span><span class="co">## 17    R    H A2         2   2 32.61      2   33.26</span></span>
<span><span class="co">## 18    R    H A2         3   2 32.56      2   33.36</span></span>
<span><span class="co">## 19    S    L A1         1   2 26.85      2   26.94</span></span>
<span><span class="co">## 20    S    L A1         2   2 28.17      2   27.69</span></span>
<span><span class="co">## 21    S    L A1         3   2 27.99      2   27.39</span></span>
<span><span class="co">## 22    S    L A2         1   2 28.71      2   29.45</span></span>
<span><span class="co">## 23    S    L A2         2   2 29.01      2   29.46</span></span>
<span><span class="co">## 24    S    L A2         3   2 28.82      2   29.48</span></span>
<span><span class="co">## 25    S    M A1         1   2 30.41      2   28.70</span></span>
<span><span class="co">## 26    S    M A1         2   2 29.49      2   28.66</span></span>
<span><span class="co">## 27    S    M A1         3   2 29.98      2   28.71</span></span>
<span><span class="co">## 28    S    M A2         1   2 28.91      2   28.09</span></span>
<span><span class="co">## 29    S    M A2         2   2 28.60      2   28.65</span></span>
<span><span class="co">## 30    S    M A2         3   2 28.59      2   28.37</span></span>
<span><span class="co">## 31    S    H A1         1   2 29.03      2   30.61</span></span>
<span><span class="co">## 32    S    H A1         2   2 28.73      2   30.20</span></span>
<span><span class="co">## 33    S    H A1         3   2 28.83      2   30.49</span></span>
<span><span class="co">## 34    S    H A2         1   2 28.29      2   30.84</span></span>
<span><span class="co">## 35    S    H A2         2   2 28.53      2   30.65</span></span>
<span><span class="co">## 36    S    H A2         3   2 28.28      2   30.74</span></span></code></pre>
<div class="section level4">
<h4 id="output-table-of-the-analysis">Output table of the analysis<a class="anchor" aria-label="anchor" href="#output-table-of-the-analysis"></a>
</h4>
<p>The <code>qpcrANOVARE</code> function produces the main analysis
output including mean wDCt, LCL, UCL, grouping letters, and standard
deviations for relative expression values. The standard deviation for
each treatment is derived from the biological replicates of
back-transformed wDCt data.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If the data include technical replicates, means of technical replicates</span></span>
<span><span class="co"># should be calculated first using meanTech function.</span></span>
<span></span>
<span><span class="co"># Applying ANOVA analysis</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_2factor</span>,</span>
<span>                   numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                   block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T          5 20.5489  4.1098  26.267 4.515e-06 ***</span></span>
<span><span class="co">## Residuals 12  1.8775  0.1565                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##   Genotype Drought     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1        S      D2 2.9690 2.1030 4.1918 0.0551   2.8577   3.0846    a   </span></span>
<span><span class="co">## 2        R      D2 0.9885 0.7002 1.3956 0.0841   0.9325   1.0478     b  </span></span>
<span><span class="co">## 3        S      D0 0.7955 0.5635 1.1232 0.2128   0.6864   0.9219     b  </span></span>
<span><span class="co">## 4        R      D1 0.6271 0.4441 0.8853 0.4388   0.4626   0.8500     bc </span></span>
<span><span class="co">## 5        S      D1 0.4147 0.2937 0.5854 0.2540   0.3478   0.4945      cd</span></span>
<span><span class="co">## 6        R      D0 0.2852 0.2020 0.4026 0.0208   0.2811   0.2893       d</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">Result</span></span></code></pre></div>
<pre><code><span><span class="co">##   Genotype Drought     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1        S      D2 2.9690 2.1030 4.1918 0.0551   2.8577   3.0846    a   </span></span>
<span><span class="co">## 2        R      D2 0.9885 0.7002 1.3956 0.0841   0.9325   1.0478     b  </span></span>
<span><span class="co">## 3        S      D0 0.7955 0.5635 1.1232 0.2128   0.6864   0.9219     b  </span></span>
<span><span class="co">## 4        R      D1 0.6271 0.4441 0.8853 0.4388   0.4626   0.8500     bc </span></span>
<span><span class="co">## 5        S      D1 0.4147 0.2937 0.5854 0.2540   0.3478   0.4945      cd</span></span>
<span><span class="co">## 6        R      D0 0.2852 0.2020 0.4026 0.0208   0.2811   0.2893       d</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">Post_hoc_Test</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Before plotting, the result needs to be extracted as below:</span></span>
<span><span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_1factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Result</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df Sum Sq Mean Sq F value   Pr(&gt;F)   </span></span>
<span><span class="co">## T          2 4.9393 2.46963  12.345 0.007473 **</span></span>
<span><span class="co">## Residuals  6 1.2003 0.20006                    </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##   SA     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1 L3 0.9885 0.6379 1.5318 0.0841   0.9325   1.0478      a </span></span>
<span><span class="co">## 2 L2 0.6271 0.4047 0.9717 0.4388   0.4626   0.8500      a </span></span>
<span><span class="co">## 3 L1 0.2852 0.1840 0.4419 0.0208   0.2811   0.2893       b</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/oneFACTORplot.html">oneFACTORplot</a></span><span class="op">(</span><span class="va">out2</span>,</span>
<span>              width <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>              fill <span class="op">=</span> <span class="st">"skyblue"</span>,</span>
<span>              y.axis.adjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>              y.axis.by <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              errorbar <span class="op">=</span> <span class="st">"ci"</span>,</span>
<span>              show.letters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              letter.position.adjust <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>              ylab <span class="op">=</span> <span class="st">"Relative Expression"</span>,</span>
<span>              xlab <span class="op">=</span> <span class="st">"Factor Levels"</span>,</span>
<span>              fontsize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>              fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">addline_format</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">...</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'\\s'</span>,<span class="st">'\n'</span>,<span class="va">x</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">f2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVAFC.html">qpcrANOVAFC</a></span><span class="op">(</span><span class="va">data_1factor</span>,</span>
<span>                 numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 mainFactor.column <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 mainFactor.level.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L1"</span>,<span class="st">"L2"</span>,<span class="st">"L3"</span><span class="op">)</span>,</span>
<span>                 width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                 fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"skyblue"</span>,<span class="st">"#79CDCD"</span><span class="op">)</span>,</span>
<span>                 y.axis.by <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 letter.position.adjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                 y.axis.adjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 ylab <span class="op">=</span> <span class="st">"Fold Change"</span>,</span>
<span>                 fontsize <span class="op">=</span> <span class="fl">12</span>, plot <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                 x.axis.labels.rename <span class="op">=</span> <span class="fu">addline_format</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, </span>
<span>                                                       <span class="st">"Treatment_1 vs Control"</span>, </span>
<span>                                                       <span class="st">"Treatment_2 vs Control"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ANOVA table </span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## SA 4.9393  2.4696     2     4  14.319 0.01502 *</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ANCOVA table</span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## SA 4.9393  2.4696     2     4  14.319 0.01502 *</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##                   contrast     FC pvalue sig    LCL    UCL     se Lower.se</span></span>
<span><span class="co">## 1                  Control 1.0000 1.0000     0.0000 0.0000 0.0208   0.9857</span></span>
<span><span class="co">## 2 Treatment_1\nvs\nControl 2.1987 0.0285   * 0.9514 5.0812 0.4388   1.6221</span></span>
<span><span class="co">## 3 Treatment_2\nvs\nControl 3.4661 0.0061  ** 1.4999 8.0101 0.0841   3.2698</span></span>
<span><span class="co">##   Upper.se</span></span>
<span><span class="co">## 1   1.0145</span></span>
<span><span class="co">## 2   2.9803</span></span>
<span><span class="co">## 3   3.6742</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/multiplot.html">multiplot</a></span><span class="op">(</span><span class="va">f1</span>, <span class="va">f2</span><span class="op">$</span><span class="va">FC_Plot_of_the_main_factor_levels</span>, cols <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"A"</span>, x <span class="op">=</span> <span class="fl">0.02</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"B"</span>, x <span class="op">=</span> <span class="fl">0.52</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-11-1.png" alt="A: bar plot representing Relative expression of a gene under three levels of a factor generated using `oneFACTORplot` function, B: Plot of the Fold change expression produced by the `qpcrANOVAFC` function from the same data used for 'A'. The first element in the `mainFactor.level.order` argument (here L1) is served as the Reference level, although the x-axis names have later been renamed by the `x.axis.labels.rename` argument. Error bars represent 95% confidence interval in A and standard error in B." width="864"><p class="caption">
A: bar plot representing Relative expression of a gene under three
levels of a factor generated using <code>oneFACTORplot</code> function,
B: Plot of the Fold change expression produced by the
<code>qpcrANOVAFC</code> function from the same data used for ‘A’. The
first element in the <code>mainFactor.level.order</code> argument (here
L1) is served as the Reference level, although the x-axis names have
later been renamed by the <code>x.axis.labels.rename</code> argument.
Error bars represent 95% confidence interval in A and standard error in
B.
</p>
</div>
</div>
<div class="section level4">
<h4 id="barplot-with-the-1-alpha-confidence-interval-as-error-bars">Barplot with the (1-alpha)% confidence interval as error bars<a class="anchor" aria-label="anchor" href="#barplot-with-the-1-alpha-confidence-interval-as-error-bars"></a>
</h4>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Before plotting, the result needs to be extracted as below:</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_2factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Result</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T          5 20.5489  4.1098  26.267 4.515e-06 ***</span></span>
<span><span class="co">## Residuals 12  1.8775  0.1565                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##   Genotype Drought     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1        S      D2 2.9690 2.1030 4.1918 0.0551   2.8577   3.0846    a   </span></span>
<span><span class="co">## 2        R      D2 0.9885 0.7002 1.3956 0.0841   0.9325   1.0478     b  </span></span>
<span><span class="co">## 3        S      D0 0.7955 0.5635 1.1232 0.2128   0.6864   0.9219     b  </span></span>
<span><span class="co">## 4        R      D1 0.6271 0.4441 0.8853 0.4388   0.4626   0.8500     bc </span></span>
<span><span class="co">## 5        S      D1 0.4147 0.2937 0.5854 0.2540   0.3478   0.4945      cd</span></span>
<span><span class="co">## 6        R      D0 0.2852 0.2020 0.4026 0.0208   0.2811   0.2893       d</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Final_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_2factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Final_data</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T          5 20.5489  4.1098  26.267 4.515e-06 ***</span></span>
<span><span class="co">## Residuals 12  1.8775  0.1565                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##   Genotype Drought     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1        S      D2 2.9690 2.1030 4.1918 0.0551   2.8577   3.0846    a   </span></span>
<span><span class="co">## 2        R      D2 0.9885 0.7002 1.3956 0.0841   0.9325   1.0478     b  </span></span>
<span><span class="co">## 3        S      D0 0.7955 0.5635 1.1232 0.2128   0.6864   0.9219     b  </span></span>
<span><span class="co">## 4        R      D1 0.6271 0.4441 0.8853 0.4388   0.4626   0.8500     bc </span></span>
<span><span class="co">## 5        S      D1 0.4147 0.2937 0.5854 0.2540   0.3478   0.4945      cd</span></span>
<span><span class="co">## 6        R      D0 0.2852 0.2020 0.4026 0.0208   0.2811   0.2893       d</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot of the 'res' data with 'Genotype' as grouping factor</span></span>
<span><span class="va">q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/twoFACTORplot.html">twoFACTORplot</a></span><span class="op">(</span><span class="va">res</span>,</span>
<span>   x.axis.factor <span class="op">=</span> <span class="va">Drought</span>,</span>
<span>   group.factor <span class="op">=</span> <span class="va">Genotype</span>,</span>
<span>   errorbar <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>   width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>   fill <span class="op">=</span> <span class="st">"Greens"</span>,</span>
<span>   y.axis.adjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>   y.axis.by <span class="op">=</span> <span class="fl">2</span>,</span>
<span>   ylab <span class="op">=</span> <span class="st">"Relative Expression"</span>,</span>
<span>   xlab <span class="op">=</span> <span class="st">"Drought Levels"</span>,</span>
<span>   legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>   show.letters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>   fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plotting the same data with 'Drought' as grouping factor</span></span>
<span><span class="va">q2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/twoFACTORplot.html">twoFACTORplot</a></span><span class="op">(</span><span class="va">res</span>,</span>
<span>   x.axis.factor <span class="op">=</span> <span class="va">Genotype</span>,</span>
<span>   group.factor <span class="op">=</span> <span class="va">Drought</span>,</span>
<span>   errorbar <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>   xlab <span class="op">=</span> <span class="st">"Genotype"</span>,</span>
<span>   fill <span class="op">=</span> <span class="st">"Blues"</span>,</span>
<span>   legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>   show.letters <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>   show.errorbars <span class="op">=</span> <span class="cn">T</span>,</span>
<span>   fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/multiplot.html">multiplot</a></span><span class="op">(</span><span class="va">q1</span>, <span class="va">q2</span>, cols <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"A"</span>, x <span class="op">=</span> <span class="fl">0.02</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"B"</span>, x <span class="op">=</span> <span class="fl">0.52</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-12-1.png" alt="Relative expression of a target gene under two different factors of genotype (with two levels) and drought (with three levels). Error bars represent standard error. Means (columns) lacking letters in common have significant difference at alpha = 0.05 as resulted from a `LSD.test`." width="864"><p class="caption">
Relative expression of a target gene under two different factors of
genotype (with two levels) and drought (with three levels). Error bars
represent standard error. Means (columns) lacking letters in common have
significant difference at alpha = 0.05 as resulted from a
<code>LSD.test</code>.
</p>
</div>
</div>
<div class="section level4">
<h4 id="a-three-factorial-experiment-example">A three-factorial experiment example<a class="anchor" aria-label="anchor" href="#a-three-factorial-experiment-example"></a>
</h4>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Before plotting, the result needs to be extracted as below:</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_3factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Result</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T         11 94.001  8.5456  29.188 3.248e-11 ***</span></span>
<span><span class="co">## Residuals 24  7.027  0.2928                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##    Type Conc SA     RE    LCL    UCL     se Lower.se Upper.se  letters</span></span>
<span><span class="co">## 1     S    H A2 5.1934 3.3217 8.1197 0.1309   4.7429   5.6867  a      </span></span>
<span><span class="co">## 2     S    H A1 2.9690 1.8990 4.6420 0.0551   2.8577   3.0846  ab     </span></span>
<span><span class="co">## 3     R    H A2 1.7371 1.1110 2.7159 0.0837   1.6392   1.8409   bc    </span></span>
<span><span class="co">## 4     S    L A2 1.5333 0.9807 2.3973 0.0865   1.4441   1.6280    c    </span></span>
<span><span class="co">## 5     R    H A1 0.9885 0.6323 1.5455 0.0841   0.9325   1.0478    cd   </span></span>
<span><span class="co">## 6     S    L A1 0.7955 0.5088 1.2438 0.2128   0.6864   0.9219     d   </span></span>
<span><span class="co">## 7     S    M A2 0.7955 0.5088 1.2438 0.2571   0.6656   0.9507     d   </span></span>
<span><span class="co">## 8     R    M A1 0.6271 0.4011 0.9804 0.4388   0.4626   0.8500     de  </span></span>
<span><span class="co">## 9     S    M A1 0.4147 0.2652 0.6483 0.2540   0.3478   0.4945      ef </span></span>
<span><span class="co">## 10    R    M A2 0.3150 0.2015 0.4925 0.2890   0.2578   0.3849       f </span></span>
<span><span class="co">## 11    R    L A1 0.2852 0.1824 0.4459 0.0208   0.2811   0.2893       f </span></span>
<span><span class="co">## 12    R    L A2 0.0641 0.0410 0.1002 0.8228   0.0362   0.1134        g</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span></span></code></pre></div>
<pre><code><span><span class="co">##    Type Conc SA     RE    LCL    UCL     se Lower.se Upper.se  letters</span></span>
<span><span class="co">## 1     S    H A2 5.1934 3.3217 8.1197 0.1309   4.7429   5.6867  a      </span></span>
<span><span class="co">## 2     S    H A1 2.9690 1.8990 4.6420 0.0551   2.8577   3.0846  ab     </span></span>
<span><span class="co">## 3     R    H A2 1.7371 1.1110 2.7159 0.0837   1.6392   1.8409   bc    </span></span>
<span><span class="co">## 4     S    L A2 1.5333 0.9807 2.3973 0.0865   1.4441   1.6280    c    </span></span>
<span><span class="co">## 5     R    H A1 0.9885 0.6323 1.5455 0.0841   0.9325   1.0478    cd   </span></span>
<span><span class="co">## 6     S    L A1 0.7955 0.5088 1.2438 0.2128   0.6864   0.9219     d   </span></span>
<span><span class="co">## 7     S    M A2 0.7955 0.5088 1.2438 0.2571   0.6656   0.9507     d   </span></span>
<span><span class="co">## 8     R    M A1 0.6271 0.4011 0.9804 0.4388   0.4626   0.8500     de  </span></span>
<span><span class="co">## 9     S    M A1 0.4147 0.2652 0.6483 0.2540   0.3478   0.4945      ef </span></span>
<span><span class="co">## 10    R    M A2 0.3150 0.2015 0.4925 0.2890   0.2578   0.3849       f </span></span>
<span><span class="co">## 11    R    L A1 0.2852 0.1824 0.4459 0.0208   0.2811   0.2893       f </span></span>
<span><span class="co">## 12    R    L A2 0.0641 0.0410 0.1002 0.8228   0.0362   0.1134        g</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># releveling a factor levels first</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">Conc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">Conc</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L"</span>,<span class="st">"M"</span>,<span class="st">"H"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">Type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">Type</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S"</span>,<span class="st">"R"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Arrange the first three colunms of the result table.</span></span>
<span><span class="co"># This determines the columns order and shapes the plot output.</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/threeFACTORplot.html">threeFACTORplot</a></span><span class="op">(</span><span class="va">res</span>,</span>
<span>    arrangement <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>    errorbar <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.85</span><span class="op">)</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>    fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># When using ci as error, increase y.axis.adjust to see the plot correctly!</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/threeFACTORplot.html">threeFACTORplot</a></span><span class="op">(</span><span class="va">res</span>,</span>
<span>   arrangement <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>   bar.width <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>   fill <span class="op">=</span> <span class="st">"Greens"</span>,</span>
<span>   xlab <span class="op">=</span> <span class="st">"Drought"</span>,</span>
<span>   ylab <span class="op">=</span> <span class="st">"Relative Expression"</span>,</span>
<span>   errorbar <span class="op">=</span> <span class="st">"ci"</span>,</span>
<span>   y.axis.adjust <span class="op">=</span> <span class="fl">2</span>,</span>
<span>   y.axis.by <span class="op">=</span> <span class="fl">2</span>,</span>
<span>   letter.position.adjust <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>   legend.title <span class="op">=</span> <span class="st">"Genotype"</span>,</span>
<span>   fontsize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>   legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>   show.letters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>   fontsizePvalue <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/multiplot.html">multiplot</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, cols <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"A"</span>, x <span class="op">=</span> <span class="fl">0.02</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">grid.text</span><span class="op">(</span><span class="st">"B"</span>, x <span class="op">=</span> <span class="fl">0.52</span>, y <span class="op">=</span> <span class="fl">1</span>, just <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>, <span class="st">"top"</span><span class="op">)</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">16</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-13-1.png" alt="A and B) Relative expression (RE) of a target gene from a three-factorial experiment data produced by  `threeFACTORplot`  function. Error bars represent standard error (A), although can be set to confidence interval (B). Means (columns) lacking letters in common have significant differences at alpha = 0.05 as resulted from an ‘LSD.test’." width="1056"><p class="caption">
A and B) Relative expression (RE) of a target gene from a
three-factorial experiment data produced by <code>threeFACTORplot</code>
function. Error bars represent standard error (A), although can be set
to confidence interval (B). Means (columns) lacking letters in common
have significant differences at alpha = 0.05 as resulted from an
‘LSD.test’.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="repeated-measure-data-in-qpcr">Repeated measure data in qpcr<a class="anchor" aria-label="anchor" href="#repeated-measure-data-in-qpcr"></a>
</h3>
<p>Fold change (FC) analysis of observations repeatedly taken over time
<code>qpcrREPEATED</code> function, for Repeated measure analysis of
uni- or multi-factorial experiment data. The bar plot of the fold
changes (FC) values along with the standard error (se) or confidence
interval (ci) is also returned by the <code>qpcrREPEATED</code>
function.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_1</span>,</span>
<span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#778899"</span>, <span class="st">"#BCD2EE"</span><span class="op">)</span>,</span>
<span>                  factor <span class="op">=</span> <span class="st">"time"</span>,</span>
<span>                  axis.text.x.angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>                  axis.text.x.hjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrREPEATED(data_repeated_measure_1, numberOfrefGenes = 1, block =</span></span>
<span><span class="co">## NULL, : The level 1 of the selected factor was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##      Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time 11.073  5.5364     2     4  4.5382 0.09357 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484</span></span>
<span><span class="co">## 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841</span></span>
<span><span class="co">## 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_2</span>,</span>
<span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  factor <span class="op">=</span> <span class="st">"time"</span>,</span>
<span>                  block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  axis.text.x.angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>                  axis.text.x.hjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrREPEATED(data_repeated_measure_2, numberOfrefGenes = 1, factor =</span></span>
<span><span class="co">## "time", : The level 1 of the selected factor was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## NOTE: Results may be misleading due to involvement in interactions</span></span></code></pre>
<pre><code><span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##                Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time           5.9166 2.95832     2     8  3.4888 0.08139 .</span></span>
<span><span class="co">## treatment      0.6773 0.67728     1     4  0.7987 0.42199  </span></span>
<span><span class="co">## time:treatment 6.3186 3.15932     2     8  3.7258 0.07186 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000 0.0000 0.7036   0.6140   1.6286</span></span>
<span><span class="co">## 2 time2 vs time1 1.2556 0.5540     0.4381 3.5987 0.5323   0.8682   1.8159</span></span>
<span><span class="co">## 3 time3 vs time1 2.5432 0.0351   * 0.8873 7.2895 0.7074   1.5575   4.1527</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/multiplot.html">multiplot</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">b</span>, cols <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-14-1.png" alt="Fold change expression (FC) of a target gene from a one and a two factorial experiment data produced by  `qpcrREPEATED`  function. Error bars represent standard error (A), although can be set to confidence interval." width="672"><p class="caption">
Fold change expression (FC) of a target gene from a one and a two
factorial experiment data produced by <code>qpcrREPEATED</code>
function. Error bars represent standard error (A), although can be set
to confidence interval.
</p>
</div>
<pre><code><span><span class="co">## $Final_data</span></span>
<span><span class="co">##   id time Etarget Cttarget Eref Ctref   wDCt</span></span>
<span><span class="co">## 1  1    1       2    18.92    2 32.77 -13.85</span></span>
<span><span class="co">## 2  1    2       2    19.46    2 33.03 -13.57</span></span>
<span><span class="co">## 3  1    3       2    15.73    2 32.95 -17.22</span></span>
<span><span class="co">## 4  2    1       2    15.82    2 32.45 -16.63</span></span>
<span><span class="co">## 5  2    2       2    17.56    2 33.24 -15.68</span></span>
<span><span class="co">## 6  2    3       2    17.21    2 33.64 -16.43</span></span>
<span><span class="co">## 7  3    1       2    19.84    2 31.62 -11.78</span></span>
<span><span class="co">## 8  3    2       2    19.74    2 32.08 -12.34</span></span>
<span><span class="co">## 9  3    3       2    18.09    2 33.40 -15.31</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $lm</span></span>
<span><span class="co">## Linear mixed model fit by REML ['lmerModLmerTest']</span></span>
<span><span class="co">## Formula: formula</span></span>
<span><span class="co">##    Data: x</span></span>
<span><span class="co">## REML criterion at convergence: 25.0824</span></span>
<span><span class="co">## Random effects:</span></span>
<span><span class="co">##  Groups   Name        Std.Dev.</span></span>
<span><span class="co">##  id       (Intercept) 1.419   </span></span>
<span><span class="co">##  Residual             1.105   </span></span>
<span><span class="co">## Number of obs: 9, groups:  id, 3</span></span>
<span><span class="co">## Fixed Effects:</span></span>
<span><span class="co">## (Intercept)        time2        time3  </span></span>
<span><span class="co">##    -14.0867       0.2233      -2.2333  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $ANOVA_table</span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##      Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time 11.073  5.5364     2     4  4.5382 0.09357 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $FC_statistics_of_the_main_factor</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484</span></span>
<span><span class="co">## 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841</span></span>
<span><span class="co">## 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $FC_Plot</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attr(,"class")</span></span>
<span><span class="co">## [1] "XX"</span></span>
<span><span class="co">## $Final_data</span></span>
<span><span class="co">##    id treatment time Etarget Cttarget Eref Ctref   wDCt</span></span>
<span><span class="co">## 1   1 untreated    1       2    19.24    2 33.73 -14.49</span></span>
<span><span class="co">## 2   1 untreated    2       2    19.95    2 34.20 -14.25</span></span>
<span><span class="co">## 3   1 untreated    3       2    19.16    2 33.90 -14.74</span></span>
<span><span class="co">## 4   2 untreated    1       2    20.11    2 32.56 -12.45</span></span>
<span><span class="co">## 5   2 untreated    2       2    20.91    2 33.98 -13.07</span></span>
<span><span class="co">## 6   2 untreated    3       2    20.91    2 33.16 -12.25</span></span>
<span><span class="co">## 7   3 untreated    1       2    20.63    2 33.72 -13.09</span></span>
<span><span class="co">## 8   3 untreated    2       2    19.16    2 34.51 -15.35</span></span>
<span><span class="co">## 9   3 untreated    3       2    19.91    2 34.33 -14.42</span></span>
<span><span class="co">## 10  4   treated    1       2    18.92    2 32.77 -13.85</span></span>
<span><span class="co">## 11  4   treated    2       2    19.46    2 33.03 -13.57</span></span>
<span><span class="co">## 12  4   treated    3       2    15.73    2 32.95 -17.22</span></span>
<span><span class="co">## 13  5   treated    1       2    15.82    2 32.45 -16.63</span></span>
<span><span class="co">## 14  5   treated    2       2    17.56    2 33.24 -15.68</span></span>
<span><span class="co">## 15  5   treated    3       2    17.21    2 33.64 -16.43</span></span>
<span><span class="co">## 16  6   treated    1       2    19.84    2 31.62 -11.78</span></span>
<span><span class="co">## 17  6   treated    2       2    19.74    2 32.08 -12.34</span></span>
<span><span class="co">## 18  6   treated    3       2    18.09    2 33.40 -15.31</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $lm</span></span>
<span><span class="co">## Linear mixed model fit by REML ['lmerModLmerTest']</span></span>
<span><span class="co">## Formula: formula</span></span>
<span><span class="co">##    Data: x</span></span>
<span><span class="co">## REML criterion at convergence: 45.9708</span></span>
<span><span class="co">## Random effects:</span></span>
<span><span class="co">##  Groups   Name        Std.Dev.</span></span>
<span><span class="co">##  id       (Intercept) 1.2134  </span></span>
<span><span class="co">##  Residual             0.9208  </span></span>
<span><span class="co">## Number of obs: 18, groups:  id, 6</span></span>
<span><span class="co">## Fixed Effects:</span></span>
<span><span class="co">##            (Intercept)                   time2                   time3  </span></span>
<span><span class="co">##               -13.3433                 -0.8800                 -0.4600  </span></span>
<span><span class="co">##       treatmenttreated  time2:treatmenttreated  time3:treatmenttreated  </span></span>
<span><span class="co">##                -0.7433                  1.1033                 -1.7733  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $ANOVA_table</span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##                Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time           5.9166 2.95832     2     8  3.4888 0.08139 .</span></span>
<span><span class="co">## treatment      0.6773 0.67728     1     4  0.7987 0.42199  </span></span>
<span><span class="co">## time:treatment 6.3186 3.15932     2     8  3.7258 0.07186 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $FC_statistics_of_the_main_factor</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000 0.0000 0.7036   0.6140   1.6286</span></span>
<span><span class="co">## 2 time2 vs time1 1.2556 0.5540     0.4381 3.5987 0.5323   0.8682   1.8159</span></span>
<span><span class="co">## 3 time3 vs time1 2.5432 0.0351   * 0.8873 7.2895 0.7074   1.5575   4.1527</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $FC_Plot</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attr(,"class")</span></span>
<span><span class="co">## [1] "XX"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="fold-change-analysis-using-a-model">Fold change analysis using a model<a class="anchor" aria-label="anchor" href="#fold-change-analysis-using-a-model"></a>
</h3>
<p>The <code>qpcrMeans</code> performs fold change <span class="math inline">\({\Delta \Delta C_t}\)</span> mwthid analysis using
a model produced by the <code>qpcrANOVAFC</code> or
<code>qpcrREPEATED.</code> The values can be returned for any effects in
the model including simple effects, interactions and slicing if an ANOVA
model is used, but ANCOVA models returned by rtpcr package only include
simple effects.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Returning fold change values from a fitted model.</span></span>
<span><span class="co"># Firstly, result of `qpcrANOVAFC` or `qpcrREPEATED` is </span></span>
<span><span class="co"># acquired which includes a model object:</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVAFC.html">qpcrANOVAFC</a></span><span class="op">(</span><span class="va">data_3factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, mainFactor.column <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrANOVAFC(data_3factor, numberOfrefGenes = 1, mainFactor.column =</span></span>
<span><span class="co">## 1, : The R level was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## NOTE: Results may be misleading due to involvement in interactions</span></span></code></pre>
<pre><code><span><span class="co">## ANOVA table </span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##              Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span></span>
<span><span class="co">## Type         24.834 24.8336     1    24 84.8207 2.392e-09 ***</span></span>
<span><span class="co">## Conc         45.454 22.7269     2    24 77.6252 3.319e-11 ***</span></span>
<span><span class="co">## SA            0.032  0.0324     1    24  0.1107 0.7422788    </span></span>
<span><span class="co">## Type:Conc    10.641  5.3203     2    24 18.1718 1.567e-05 ***</span></span>
<span><span class="co">## Type:SA       6.317  6.3168     1    24 21.5756 0.0001024 ***</span></span>
<span><span class="co">## Conc:SA       3.030  1.5150     2    24  5.1747 0.0135366 *  </span></span>
<span><span class="co">## Type:Conc:SA  3.694  1.8470     2    24  6.3086 0.0062852 ** </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ANCOVA table</span></span>
<span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##      Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span></span>
<span><span class="co">## SA    0.032  0.0324     1    31  0.0327    0.8577    </span></span>
<span><span class="co">## Conc 45.454 22.7269     2    31 22.9429 7.682e-07 ***</span></span>
<span><span class="co">## Type 24.834 24.8336     1    31 25.0696 2.105e-05 ***</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##   contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1        R 1.0000      1     0.0000 0.0000 0.3939   0.7611   1.3139</span></span>
<span><span class="co">## 2   S vs R 3.1626      0 *** 2.4433 4.0936 0.3064   2.5575   3.9109</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change plot of the main factor levels</span></span></code></pre>
<p><img src="vignette_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Returning fold change values of Conc levels from a fitted model:</span></span>
<span><span class="fu"><a href="../reference/qpcrMeans.html">qpcrMeans</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">lm_ANOVA</span>, specs <span class="op">=</span> <span class="st">"Conc"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  contrast       FC        SE df      LCL      UCL p.value sig</span></span>
<span><span class="co">##  M vs L   1.307369 0.2208988 22 0.951681 1.795994  0.0940 .  </span></span>
<span><span class="co">##  H vs L   5.869889 0.2208988 22 4.272905 8.063742  &lt;.0001 ***</span></span>
<span><span class="co">##  H vs M   4.489848 0.2208988 22 3.268323 6.167915  &lt;.0001 ***</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Results are averaged over the levels of: Type, SA </span></span>
<span><span class="co">## Degrees-of-freedom method: kenward-roger </span></span>
<span><span class="co">## Confidence level used: 0.95</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Returning fold change values of Conc levels sliced by Type*SA:</span></span>
<span><span class="fu"><a href="../reference/qpcrMeans.html">qpcrMeans</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">lm_ANOVA</span>, specs <span class="op">=</span> <span class="st">"Conc | (Type*SA)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Type = R, SA = A1:</span></span>
<span><span class="co">##  contrast        FC        SE df       LCL      UCL p.value sig</span></span>
<span><span class="co">##  M vs L    2.198724 0.4417977 22  1.165084  4.14939  0.0174 *  </span></span>
<span><span class="co">##  H vs L    3.466148 0.4417977 22  1.836681  6.54125  0.0005 ***</span></span>
<span><span class="co">##  H vs M    1.576436 0.4417977 22  0.835339  2.97502  0.1514    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Type = S, SA = A1:</span></span>
<span><span class="co">##  contrast        FC        SE df       LCL      UCL p.value sig</span></span>
<span><span class="co">##  M vs L    0.521233 0.4417977 22  0.276197  0.98366  0.0448 *  </span></span>
<span><span class="co">##  H vs L    3.732132 0.4417977 22  1.977623  7.04321  0.0003 ***</span></span>
<span><span class="co">##  H vs M    7.160201 0.4417977 22  3.794126 13.51259  &lt;.0001 ***</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Type = R, SA = A2:</span></span>
<span><span class="co">##  contrast        FC        SE df       LCL      UCL p.value sig</span></span>
<span><span class="co">##  M vs L    4.913213 0.4417977 22  2.603467  9.27212  &lt;.0001 ***</span></span>
<span><span class="co">##  H vs L   27.095850 0.4417977 22 14.357849 51.13476  &lt;.0001 ***</span></span>
<span><span class="co">##  H vs M    5.514895 0.4417977 22  2.922293 10.40760  &lt;.0001 ***</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Type = S, SA = A2:</span></span>
<span><span class="co">##  contrast        FC        SE df       LCL      UCL p.value sig</span></span>
<span><span class="co">##  M vs L    0.518830 0.4417977 22  0.274923  0.97913  0.0434 *  </span></span>
<span><span class="co">##  H vs L    3.386981 0.4417977 22  1.794731  6.39184  0.0006 ***</span></span>
<span><span class="co">##  H vs M    6.528116 0.4417977 22  3.459190 12.31973  &lt;.0001 ***</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Degrees-of-freedom method: kenward-roger </span></span>
<span><span class="co">## Confidence level used: 0.95</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Returning fold change values of Conc</span></span>
<span><span class="fu"><a href="../reference/qpcrMeans.html">qpcrMeans</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">lm_ANOVA</span>, specs <span class="op">=</span> <span class="st">"Conc * Type"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  contrast          FC        SE df       LCL      UCL p.value sig</span></span>
<span><span class="co">##  M R vs L R  3.286761 0.3123981 22  2.097677  5.14989  &lt;.0001 ***</span></span>
<span><span class="co">##  H R vs L R  9.691142 0.3123981 22  6.185081 15.18464  &lt;.0001 ***</span></span>
<span><span class="co">##  L S vs L R  8.168097 0.3123981 22  5.213044 12.79824  &lt;.0001 ***</span></span>
<span><span class="co">##  M S vs L R  4.247655 0.3123981 22  2.710939  6.65547  &lt;.0001 ***</span></span>
<span><span class="co">##  H S vs L R 29.040613 0.3123981 22 18.534303 45.50251  &lt;.0001 ***</span></span>
<span><span class="co">##  H R vs M R  2.948538 0.3123981 22  1.881816  4.61994  0.0001 ***</span></span>
<span><span class="co">##  L S vs M R  2.485151 0.3123981 22  1.586073  3.89388  0.0004 ***</span></span>
<span><span class="co">##  M S vs M R  1.292353 0.3123981 22  0.824806  2.02493  0.2489    </span></span>
<span><span class="co">##  H S vs M R  8.835632 0.3123981 22  5.639078 13.84418  &lt;.0001 ***</span></span>
<span><span class="co">##  L S vs H R  0.842842 0.3123981 22  0.537918  1.32061  0.4382    </span></span>
<span><span class="co">##  M S vs H R  0.438303 0.3123981 22  0.279734  0.68676  0.0010 ***</span></span>
<span><span class="co">##  H S vs H R  2.996614 0.3123981 22  1.912499  4.69527  &lt;.0001 ***</span></span>
<span><span class="co">##  M S vs L S  0.520030 0.3123981 22  0.331894  0.81481  0.0063 ** </span></span>
<span><span class="co">##  H S vs L S  3.555371 0.3123981 22  2.269109  5.57076  &lt;.0001 ***</span></span>
<span><span class="co">##  H S vs M S  6.836857 0.3123981 22  4.363420 10.71238  &lt;.0001 ***</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Results are averaged over the levels of: SA </span></span>
<span><span class="co">## Degrees-of-freedom method: kenward-roger </span></span>
<span><span class="co">## Confidence level used: 0.95</span></span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Returning fold change values of Conc levels sliced by Type:</span></span>
<span><span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrMeans.html">qpcrMeans</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">lm_ANOVA</span>, specs <span class="op">=</span> <span class="st">"Conc | Type"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/twoFACTORplot.html">twoFACTORplot</a></span><span class="op">(</span><span class="va">res2</span>, x.axis.factor <span class="op">=</span> <span class="va">contrast</span>, ylab <span class="op">=</span> <span class="st">"Fold Change"</span>,</span>
<span>              group.factor <span class="op">=</span> <span class="va">Type</span>, errorbar <span class="op">=</span> <span class="st">"ci"</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="an-example-of-showing-point-on-the-plot">An example of showing point on the plot<a class="anchor" aria-label="anchor" href="#an-example-of-showing-point-on-the-plot"></a>
</h2>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_3factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Result</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T         11 94.001  8.5456  29.188 3.248e-11 ***</span></span>
<span><span class="co">## Residuals 24  7.027  0.2928                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##    Type Conc SA     RE    LCL    UCL     se Lower.se Upper.se  letters</span></span>
<span><span class="co">## 1     S    H A2 5.1934 3.3217 8.1197 0.1309   4.7429   5.6867  a      </span></span>
<span><span class="co">## 2     S    H A1 2.9690 1.8990 4.6420 0.0551   2.8577   3.0846  ab     </span></span>
<span><span class="co">## 3     R    H A2 1.7371 1.1110 2.7159 0.0837   1.6392   1.8409   bc    </span></span>
<span><span class="co">## 4     S    L A2 1.5333 0.9807 2.3973 0.0865   1.4441   1.6280    c    </span></span>
<span><span class="co">## 5     R    H A1 0.9885 0.6323 1.5455 0.0841   0.9325   1.0478    cd   </span></span>
<span><span class="co">## 6     S    L A1 0.7955 0.5088 1.2438 0.2128   0.6864   0.9219     d   </span></span>
<span><span class="co">## 7     S    M A2 0.7955 0.5088 1.2438 0.2571   0.6656   0.9507     d   </span></span>
<span><span class="co">## 8     R    M A1 0.6271 0.4011 0.9804 0.4388   0.4626   0.8500     de  </span></span>
<span><span class="co">## 9     S    M A1 0.4147 0.2652 0.6483 0.2540   0.3478   0.4945      ef </span></span>
<span><span class="co">## 10    R    M A2 0.3150 0.2015 0.4925 0.2890   0.2578   0.3849       f </span></span>
<span><span class="co">## 11    R    L A1 0.2852 0.1824 0.4459 0.0208   0.2811   0.2893       f </span></span>
<span><span class="co">## 12    R    L A2 0.0641 0.0410 0.1002 0.8228   0.0362   0.1134        g</span></span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_3factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Final_data</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## T         11 94.001  8.5456  29.188 3.248e-11 ***</span></span>
<span><span class="co">## Residuals 24  7.027  0.2928                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##    Type Conc SA     RE    LCL    UCL     se Lower.se Upper.se  letters</span></span>
<span><span class="co">## 1     S    H A2 5.1934 3.3217 8.1197 0.1309   4.7429   5.6867  a      </span></span>
<span><span class="co">## 2     S    H A1 2.9690 1.8990 4.6420 0.0551   2.8577   3.0846  ab     </span></span>
<span><span class="co">## 3     R    H A2 1.7371 1.1110 2.7159 0.0837   1.6392   1.8409   bc    </span></span>
<span><span class="co">## 4     S    L A2 1.5333 0.9807 2.3973 0.0865   1.4441   1.6280    c    </span></span>
<span><span class="co">## 5     R    H A1 0.9885 0.6323 1.5455 0.0841   0.9325   1.0478    cd   </span></span>
<span><span class="co">## 6     S    L A1 0.7955 0.5088 1.2438 0.2128   0.6864   0.9219     d   </span></span>
<span><span class="co">## 7     S    M A2 0.7955 0.5088 1.2438 0.2571   0.6656   0.9507     d   </span></span>
<span><span class="co">## 8     R    M A1 0.6271 0.4011 0.9804 0.4388   0.4626   0.8500     de  </span></span>
<span><span class="co">## 9     S    M A1 0.4147 0.2652 0.6483 0.2540   0.3478   0.4945      ef </span></span>
<span><span class="co">## 10    R    M A2 0.3150 0.2015 0.4925 0.2890   0.2578   0.3849       f </span></span>
<span><span class="co">## 11    R    L A1 0.2852 0.1824 0.4459 0.0208   0.2811   0.2893       f </span></span>
<span><span class="co">## 12    R    L A2 0.0641 0.0410 0.1002 0.8228   0.0362   0.1134        g</span></span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Arrange factor levels to your desired order:</span></span>
<span><span class="va">b</span><span class="op">$</span><span class="va">Conc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">Conc</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L"</span>,<span class="st">"M"</span>,<span class="st">"H"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">a</span><span class="op">$</span><span class="va">Conc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">a</span><span class="op">$</span><span class="va">Conc</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L"</span>,<span class="st">"M"</span>,<span class="st">"H"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generating plot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">b</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Type</span>, y <span class="op">=</span> <span class="va">RE</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Conc</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">SA</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Reds"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Relative Expression"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">a</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Type</span>, y <span class="op">=</span> <span class="op">(</span><span class="fl">2</span><span class="op">^</span><span class="op">(</span><span class="op">-</span><span class="va">wDCt</span><span class="op">)</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Conc</span><span class="op">)</span><span class="op">)</span>, </span>
<span>             position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"ylab"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"xlab"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, color <span class="op">=</span> <span class="st">"black"</span>, angle <span class="op">=</span> <span class="fl">0</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, color <span class="op">=</span> <span class="st">"black"</span>, angle <span class="op">=</span> <span class="fl">0</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.title  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">RE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">se</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.1</span>, by <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, </span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">RE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">se</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.1</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## Warning: A numeric `legend.position` argument in `theme()` was deprecated in ggplot2</span></span>
<span><span class="co">## 3.5.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use the `legend.position.inside` argument of `theme()` instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<p><img src="vignette_files/figure-html/unnamed-chunk-16-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="checking-normality-of-residuals">Checking normality of residuals<a class="anchor" aria-label="anchor" href="#checking-normality-of-residuals"></a>
</h2>
<p>If the residuals from a <code>t.test</code> or an <code>lm</code> or
and <code>lmer</code> object are not normally distributed, the
significance results might be violated. In such cases, one could use
non-parametric tests such as the Mann-Whitney test (also known as the
Wilcoxon rank-sum test), <code><a href="https://rdrr.io/r/stats/wilcox.test.html" class="external-link">wilcox.test()</a></code>, which is an
alternative to <code>t.test</code>, or the <code><a href="https://rdrr.io/r/stats/kruskal.test.html" class="external-link">kruskal.test()</a></code>
test which alternative to one-way analysis of variance, to test the
difference between medians of the populations using independent samples.
However, the <code>t.test</code> function (along with the
<code>qpcrTTEST</code> function described above) includes the
<code>var.equal</code> argument. When set to <code>FALSE</code>, perform
<code>t.test</code> under the unequal variances hypothesis. Residuals
for <code>lm</code> (from <code>qpcrANOVARE</code> and
<code>qpcrANOVAFC</code> functions) and <code>lmer</code> (from
<code>qpcrREPEATED</code> function) objects can be extracted and plotted
as follow:</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">residuals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_1factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">lmCRD</span><span class="op">$</span><span class="va">residuals</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: wDCt</span></span>
<span><span class="co">##           Df Sum Sq Mean Sq F value   Pr(&gt;F)   </span></span>
<span><span class="co">## T          2 4.9393 2.46963  12.345 0.007473 **</span></span>
<span><span class="co">## Residuals  6 1.2003 0.20006                    </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Relative expression table</span></span>
<span><span class="co">##   SA     RE    LCL    UCL     se Lower.se Upper.se letters</span></span>
<span><span class="co">## 1 L3 0.9885 0.6379 1.5318 0.0841   0.9325   1.0478      a </span></span>
<span><span class="co">## 2 L2 0.6271 0.4047 0.9717 0.4388   0.4626   0.8500      a </span></span>
<span><span class="co">## 3 L1 0.2852 0.1840 0.4419 0.0208   0.2811   0.2893       b</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html" class="external-link">shapiro.test</a></span><span class="op">(</span><span class="va">residuals</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Shapiro-Wilk normality test</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  residuals</span></span>
<span><span class="co">## W = 0.84232, p-value = 0.06129</span></span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">residuals</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/qqnorm.html" class="external-link">qqnorm</a></span><span class="op">(</span><span class="va">residuals</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/qqnorm.html" class="external-link">qqline</a></span><span class="op">(</span><span class="va">residuals</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-17-1.png" alt="QQ-plot for the normality assessment of the residuals derived from `t.test` or `lm` functions." width="960"><p class="caption">
QQ-plot for the normality assessment of the residuals derived from
<code>t.test</code> or <code>lm</code> functions.
</p>
</div>
<p>For the repeated measure models, residulas can be extracted by
<code>residuals(a$lm)</code> and plotted by
<code>plot(residuals(a$lm))</code> where ‘a’ is an object created by the
<code>qpcrREPEATED</code> function.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_2</span>, </span>
<span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                  factor <span class="op">=</span> <span class="st">"time"</span>, </span>
<span>                  block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  y.axis.adjust <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrREPEATED(data_repeated_measure_2, numberOfrefGenes = 1, factor =</span></span>
<span><span class="co">## "time", : The level 1 of the selected factor was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## NOTE: Results may be misleading due to involvement in interactions</span></span></code></pre>
<pre><code><span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##                Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time           5.9166 2.95832     2     8  3.4888 0.08139 .</span></span>
<span><span class="co">## treatment      0.6773 0.67728     1     4  0.7987 0.42199  </span></span>
<span><span class="co">## time:treatment 6.3186 3.15932     2     8  3.7258 0.07186 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000 0.0000 0.7036   0.6140   1.6286</span></span>
<span><span class="co">## 2 time2 vs time1 1.2556 0.5540     0.4381 3.5987 0.5323   0.8682   1.8159</span></span>
<span><span class="co">## 3 time3 vs time1 2.5432 0.0351   * 0.8873 7.2895 0.7074   1.5575   4.1527</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change plot of the main factor levels</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-18-1.png" alt="QQ-plot for the normality assessment of the residuals derived from `t.test` or `lm` functions." width="384"><p class="caption">
QQ-plot for the normality assessment of the residuals derived from
<code>t.test</code> or <code>lm</code> functions.
</p>
</div>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/residuals.html" class="external-link">residuals</a></span><span class="op">(</span><span class="va">a</span><span class="op">$</span><span class="va">lm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          1          2          3          4          5          6          7 </span></span>
<span><span class="co">## -0.5566160  0.5633840 -0.3466160 -0.1133883  0.1466117  0.5466117  0.6700042 </span></span>
<span><span class="co">##          8          9         10         11         12         13         14 </span></span>
<span><span class="co">## -0.7099958 -0.1999958  0.3401353  0.3968019 -0.7965314 -1.2933207 -0.5666540 </span></span>
<span><span class="co">##         15         16         17         18 </span></span>
<span><span class="co">##  1.1400126  0.9531854  0.1698521 -0.3434812</span></span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/residuals.html" class="external-link">residuals</a></span><span class="op">(</span><span class="va">a</span><span class="op">$</span><span class="va">lm</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-18-2.png" alt="QQ-plot for the normality assessment of the residuals derived from `t.test` or `lm` functions." width="384"><p class="caption">
QQ-plot for the normality assessment of the residuals derived from
<code>t.test</code> or <code>lm</code> functions.
</p>
</div>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/qqnorm.html" class="external-link">qqnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/residuals.html" class="external-link">residuals</a></span><span class="op">(</span><span class="va">a</span><span class="op">$</span><span class="va">lm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/qqnorm.html" class="external-link">qqline</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/residuals.html" class="external-link">residuals</a></span><span class="op">(</span><span class="va">a</span><span class="op">$</span><span class="va">lm</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-18-3.png" alt="QQ-plot for the normality assessment of the residuals derived from `t.test` or `lm` functions." width="384"><p class="caption">
QQ-plot for the normality assessment of the residuals derived from
<code>t.test</code> or <code>lm</code> functions.
</p>
</div>
</div>
<div class="section level2">
<h2 id="mean-of-technical-replicates">Mean of technical replicates<a class="anchor" aria-label="anchor" href="#mean-of-technical-replicates"></a>
</h2>
<p>Calculating the mean of technical replicates and getting an output
table appropriate for subsequent ANOVA analysis can be done using the
<code>meanTech</code> function. For this, the input data set should
follow the column arrangement of the following example data. Grouping
columns must be specified under the <code>groups</code> argument of the
<code>meanTech</code> function.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See example input data frame:</span></span>
<span><span class="va">data_withTechRep</span></span></code></pre></div>
<pre><code><span><span class="co">##    factor1 factor2 factor3 biolrep techrep Etarget targetCt Eref  refCt</span></span>
<span><span class="co">## 1    Line1    Heat    Ctrl       1       1       2   33.346    2 31.520</span></span>
<span><span class="co">## 2    Line1    Heat    Ctrl       1       2       2   28.895    2 29.905</span></span>
<span><span class="co">## 3    Line1    Heat    Ctrl       1       3       2   28.893    2 29.454</span></span>
<span><span class="co">## 4    Line1    Heat    Ctrl       2       1       2   30.411    2 28.798</span></span>
<span><span class="co">## 5    Line1    Heat    Ctrl       2       2       2   33.390    2 31.574</span></span>
<span><span class="co">## 6    Line1    Heat    Ctrl       2       3       2   33.211    2 31.326</span></span>
<span><span class="co">## 7    Line1    Heat    Ctrl       3       1       2   33.845    2 31.759</span></span>
<span><span class="co">## 8    Line1    Heat    Ctrl       3       2       2   33.345    2 31.548</span></span>
<span><span class="co">## 9    Line1    Heat    Ctrl       3       3       2   32.500    2 31.477</span></span>
<span><span class="co">## 10   Line1    Heat   Treat       1       1       2   33.006    2 31.483</span></span>
<span><span class="co">## 11   Line1    Heat   Treat       1       2       2   32.588    2 31.902</span></span>
<span><span class="co">## 12   Line1    Heat   Treat       1       3       2   33.370    2 31.196</span></span>
<span><span class="co">## 13   Line1    Heat   Treat       2       1       2   36.820    2 31.440</span></span>
<span><span class="co">## 14   Line1    Heat   Treat       2       2       2   32.750    2 31.300</span></span>
<span><span class="co">## 15   Line1    Heat   Treat       2       3       2   32.450    2 32.597</span></span>
<span><span class="co">## 16   Line1    Heat   Treat       3       1       2   35.238    2 31.461</span></span>
<span><span class="co">## 17   Line1    Heat   Treat       3       2       2   28.532    2 30.651</span></span>
<span><span class="co">## 18   Line1    Heat   Treat       3       3       2   28.285    2 30.745</span></span></code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculating mean of technical replicates</span></span>
<span><span class="fu"><a href="../reference/meanTech.html">meanTech</a></span><span class="op">(</span><span class="va">data_withTechRep</span>, groups <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   factor1 factor2 factor3 biolrep Etarget targetCt Eref    refCt</span></span>
<span><span class="co">## 1   Line1    Heat    Ctrl       1       2 30.37800    2 30.29300</span></span>
<span><span class="co">## 2   Line1    Heat    Ctrl       2       2 32.33733    2 30.56600</span></span>
<span><span class="co">## 3   Line1    Heat    Ctrl       3       2 33.23000    2 31.59467</span></span>
<span><span class="co">## 4   Line1    Heat   Treat       1       2 32.98800    2 31.52700</span></span>
<span><span class="co">## 5   Line1    Heat   Treat       2       2 34.00667    2 31.77900</span></span>
<span><span class="co">## 6   Line1    Heat   Treat       3       2 30.68500    2 30.95233</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="combining-fc-results-of-different-genes">Combining FC results of different genes<a class="anchor" aria-label="anchor" href="#combining-fc-results-of-different-genes"></a>
</h2>
<p><code>qpcrANOVAFC</code> and <code>twoFACTORplot</code> functions
give FC results for one gene each time. you can combine FC tables of
different genes and present their bar plot by <code>twoFACTORplot</code>
function. An example has been shown bellow for two genes, however it
works for any number of genes.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_1</span>,</span>
<span>             numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>             factor <span class="op">=</span> <span class="st">"time"</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrREPEATED(data_repeated_measure_1, numberOfrefGenes = 1, factor =</span></span>
<span><span class="co">## "time", : The level 1 of the selected factor was used as calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##      Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time 11.073  5.5364     2     4  4.5382 0.09357 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484</span></span>
<span><span class="co">## 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841</span></span>
<span><span class="co">## 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change plot of the main factor levels</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-20-1.png" alt="Fold change expression of two different genes. FC tables of any number of genes can be combined and used as input data frame for `twoFACTORplot` function." width="480"><p class="caption">
Fold change expression of two different genes. FC tables of any number
of genes can be combined and used as input data frame for
<code>twoFACTORplot</code> function.
</p>
</div>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_2</span>,</span>
<span>                  factor <span class="op">=</span> <span class="st">"time"</span>,</span>
<span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qpcrREPEATED(data_repeated_measure_2, factor = "time",</span></span>
<span><span class="co">## numberOfrefGenes = 1, : The level 1 of the selected factor was used as</span></span>
<span><span class="co">## calibrator.</span></span></code></pre>
<pre><code><span><span class="co">## NOTE: Results may be misleading due to involvement in interactions</span></span></code></pre>
<pre><code><span><span class="co">## Type III Analysis of Variance Table with Satterthwaite's method</span></span>
<span><span class="co">##                Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span></span>
<span><span class="co">## time           5.9166 2.95832     2     8  3.4888 0.08139 .</span></span>
<span><span class="co">## treatment      0.6773 0.67728     1     4  0.7987 0.42199  </span></span>
<span><span class="co">## time:treatment 6.3186 3.15932     2     8  3.7258 0.07186 .</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change table</span></span>
<span><span class="co">##         contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000 0.0000 0.7036   0.6140   1.6286</span></span>
<span><span class="co">## 2 time2 vs time1 1.2556 0.5540     0.4381 3.5987 0.5323   0.8682   1.8159</span></span>
<span><span class="co">## 3 time3 vs time1 2.5432 0.0351   * 0.8873 7.2895 0.7074   1.5575   4.1527</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Fold Change plot of the main factor levels</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-20-2.png" alt="Fold change expression of two different genes. FC tables of any number of genes can be combined and used as input data frame for `twoFACTORplot` function." width="480"><p class="caption">
Fold change expression of two different genes. FC tables of any number
of genes can be combined and used as input data frame for
<code>twoFACTORplot</code> function.
</p>
</div>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a1</span> <span class="op">&lt;-</span> <span class="va">a</span><span class="op">$</span><span class="va">FC_statistics_of_the_main_factor</span></span>
<span><span class="va">b1</span> <span class="op">&lt;-</span> <span class="va">b</span><span class="op">$</span><span class="va">FC_statistics_of_the_main_factor</span></span>
<span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">a1</span>, <span class="va">b1</span><span class="op">)</span></span>
<span><span class="va">c</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">c</span></span></code></pre></div>
<pre><code><span><span class="co">##         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se gene</span></span>
<span><span class="co">## 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484    1</span></span>
<span><span class="co">## 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841    1</span></span>
<span><span class="co">## 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040    1</span></span>
<span><span class="co">## 4          time1 1.0000 1.0000     0.0000  0.0000 0.7036   0.6140   1.6286    2</span></span>
<span><span class="co">## 5 time2 vs time1 1.2556 0.5540     0.4381  3.5987 0.5323   0.8682   1.8159    2</span></span>
<span><span class="co">## 6 time3 vs time1 2.5432 0.0351   * 0.8873  7.2895 0.7074   1.5575   4.1527    2</span></span></code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/twoFACTORplot.html">twoFACTORplot</a></span><span class="op">(</span><span class="va">c</span>, x.axis.factor <span class="op">=</span> <span class="va">contrast</span>, </span>
<span>              group.factor <span class="op">=</span> <span class="va">gene</span>, fill <span class="op">=</span> <span class="st">'Reds'</span>, errorbar <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>              ylab <span class="op">=</span> <span class="st">"FC"</span>, axis.text.x.angle <span class="op">=</span> <span class="fl">45</span>, y.axis.adjust <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>              axis.text.x.hjust <span class="op">=</span> <span class="fl">1</span>, legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="vignette_files/figure-html/unnamed-chunk-20-3.png" alt="Fold change expression of two different genes. FC tables of any number of genes can be combined and used as input data frame for `twoFACTORplot` function." width="480"><p class="caption">
Fold change expression of two different genes. FC tables of any number
of genes can be combined and used as input data frame for
<code>twoFACTORplot</code> function.
</p>
</div>
</div>
<div class="section level2">
<h2 id="how-to-edit-ouptput-graphs">How to edit ouptput graphs?<a class="anchor" aria-label="anchor" href="#how-to-edit-ouptput-graphs"></a>
</h2>
<p>the rtpcr graphical functions create a list containing the ggplot
object so for editing or adding new layers to the graph output, you need
to extract the ggplot object first:</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qpcrANOVAFC.html">qpcrANOVAFC</a></span><span class="op">(</span><span class="va">data_2factor</span>, </span>
<span>            numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span>
<span>            mainFactor.column <span class="op">=</span> <span class="fl">1</span>,</span>
<span>            block <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            mainFactor.level.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S"</span>, <span class="st">"R"</span><span class="op">)</span>,</span>
<span>            fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#CDC673"</span>, <span class="st">"#EEDD82"</span><span class="op">)</span>,</span>
<span>            analysisType <span class="op">=</span> <span class="st">"ancova"</span>,</span>
<span>            fontsizePvalue <span class="op">=</span> <span class="fl">7</span>,</span>
<span>            y.axis.adjust <span class="op">=</span> <span class="fl">0.1</span>, width <span class="op">=</span> <span class="fl">0.35</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">b</span><span class="op">$</span><span class="va">FC_Plot_of_the_main_factor_levels</span></span>
<span><span class="va">p2</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"rtpcr"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>Email: gh.mirzaghaderi at uok.ac.ir</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of Relative
Gene Expression Data Using Real-Time Quantitative PCR and the Double
Delta CT Method. Methods 25 (4).
<a href="https://doi.org/10.1006/meth.2001.1262" class="external-link">doi.org/10.1006/meth.2001.1262</a>.</p>
<p>Ganger, MT, Dietz GD, Ewing SJ. 2017. A common base method for
analysis of qPCR data and the application of simple blocking in qPCR
experiments. BMC bioinformatics 18, 1-11.
<a href="https://doi.org/10.1186/s12859-017-1949-5" class="external-link">doi.org/10.1186/s12859-017-1949-5</a>.</p>
<p>Pfaffl MW, Horgan GW, Dempfle L. 2002. Relative expression software
tool (REST©) for group-wise comparison and statistical analysis of
relative expression results in real-time PCR. Nucleic acids research 30,
e36-e36.
<a href="https://doi.org/10.1093/nar/30.9.e36" class="external-link">doi.org/10.1093/nar/30.9.e36</a>.</p>
<p>Taylor SC, Nadeau K, Abbasi M, Lachance C, Nguyen M, Fenrich, J.
2019. The ultimate qPCR experiment: producing publication quality,
reproducible data the first time. Trends in Biotechnology, 37(7),
761-774<a href="https://doi.org/10.1016/j.tibtech.2018.12.002" class="external-link">doi.org/10.1016/j.tibtech.2018.12.002</a>.</p>
<p>Yuan, JS, Ann Reed, Feng Chen, and Neal Stewart. 2006. Statistical
Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85).
<a href="https://doi.org/10.1186/1471-2105-7-85" class="external-link">doi.org/10.1186/1471-2105-7-85</a>.</p>
<p>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ghader Mirzaghaderi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
