<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Fold change (\(\Delta \Delta C_T\) method) analysis using ANOVA and ANCOVA — qpcrANOVAFC • rtpcr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Fold change (\(\Delta \Delta C_T\) method) analysis using ANOVA and ANCOVA — qpcrANOVAFC"><meta property="og:description" content="Fold change (\(\Delta \Delta C_T\) method) analysis of qPCR data can be done using 
ANOVA (analysis of variance) and ANCOVA (analysis of covariance) by the 
qpcrANOVAFC function, for uni- or multi-factorial experiment data. The bar plot of the fold changes (FC) 
values along with the standard error (se) and confidence interval (ci) is returned by 
the qpcrANOVAFC function."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rtpcr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/vignette.html">rtpcr: a package for statistical analysis of real-time PCR data in R</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mirzaghaderi/rtpcr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Fold change (\(\Delta \Delta C_T\) method) analysis using ANOVA and ANCOVA</h1>
    <small class="dont-index">Source: <a href="https://github.com/mirzaghaderi/rtpcr/blob/HEAD/R/qpcrANOVAFC.r" class="external-link"><code>R/qpcrANOVAFC.r</code></a></small>
    <div class="hidden name"><code>qpcrANOVAFC.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Fold change (\(\Delta \Delta C_T\) method) analysis of qPCR data can be done using 
ANOVA (analysis of variance) and ANCOVA (analysis of covariance) by the 
<code>qpcrANOVAFC</code> function, for uni- or multi-factorial experiment data. The bar plot of the fold changes (FC) 
values along with the standard error (se) and confidence interval (ci) is returned by 
the <code>qpcrANOVAFC</code> function.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">numberOfrefGenes</span>,</span>
<span>  <span class="va">mainFactor.column</span>,</span>
<span>  analysisType <span class="op">=</span> <span class="st">"anova"</span>,</span>
<span>  mainFactor.level.order <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">block</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"#BFEFFF"</span>,</span>
<span>  y.axis.adjust <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  y.axis.by <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  letter.position.adjust <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Fold Change"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  fontsize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  fontsizePvalue <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  axis.text.x.angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  axis.text.x.hjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  x.axis.labels.rename <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  p.adj <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  errorbar <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>a data frame of condition(s), biological replicates, efficiency (E) and Ct values of 
target and reference genes. Each Ct value in the data frame is the mean of technical replicates. 
<strong>NOTE:</strong> Each line belongs to a separate individual reflecting a non-repeated measure experiment). 
See <a href="../doc/vignette.html"><code>vignette</code></a>, section "data structure and column arrangement" for details.</p></dd>


<dt>numberOfrefGenes</dt>
<dd><p>number of reference genes which is 1 or 2 (up to two reference genes can be handled).</p></dd>


<dt>mainFactor.column</dt>
<dd><p>the factor for which fold change expression is calculated for its levels. 
If <code>ancova</code> is selected as <code>analysisType</code>, the remaining factors (if any) are considered as covariate(s).</p></dd>


<dt>analysisType</dt>
<dd><p>should be one of "ancova" or "anova". Default is "anova".</p></dd>


<dt>mainFactor.level.order</dt>
<dd><p>NULL (default) or a vector of main factor level names. If <code>NULL</code>, 
the first level of the <code>mainFactor.column</code> is used 
as calibrator. If a vector of main factor levels (in any order) is specified, the first level in the vector is 
used as calibrator. Calibrator is the reference level or sample that all others are compared to. Examples are untreated 
of time 0. The FC value of the reference or calibrator level is 1 because it is not changed compared to itself.</p></dd>


<dt>block</dt>
<dd><p>column name of the block if there is a blocking factor (for correct column arrangement see 
example data.). When a qPCR experiment is done in multiple qPCR plates, variation resulting from the 
plates may interfere with the actual amount of gene expression. One solution is to conduct each plate 
as a complete randomized block so that at least one replicate of each treatment and control is present 
on a plate. Block effect is usually considered as random and its interaction with any main effect 
is not considered.</p></dd>


<dt>width</dt>
<dd><p>a positive number determining bar width.</p></dd>


<dt>fill</dt>
<dd><p>specify the fill color for the columns in the bar plot. If a vector of two colors is specified, 
the reference level is differentialy colored.</p></dd>


<dt>y.axis.adjust</dt>
<dd><p>a negative or positive value for reducing or increasing the length of the y axis.</p></dd>


<dt>y.axis.by</dt>
<dd><p>determines y axis step length</p></dd>


<dt>letter.position.adjust</dt>
<dd><p>adjust the distance between the signs and the error bars.</p></dd>


<dt>ylab</dt>
<dd><p>the title of the y axis</p></dd>


<dt>xlab</dt>
<dd><p>the title of the x axis</p></dd>


<dt>fontsize</dt>
<dd><p>font size of the plot</p></dd>


<dt>fontsizePvalue</dt>
<dd><p>font size of the pvalue labels</p></dd>


<dt>axis.text.x.angle</dt>
<dd><p>angle of x axis text</p></dd>


<dt>axis.text.x.hjust</dt>
<dd><p>horizontal justification of x axis text</p></dd>


<dt>x.axis.labels.rename</dt>
<dd><p>a vector replacing the x axis labels in the bar plot</p></dd>


<dt>p.adj</dt>
<dd><p>Method for adjusting p values</p></dd>


<dt>errorbar</dt>
<dd><p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p></dd>


<dt>plot</dt>
<dd><p>if <code>FALSE</code>, prevents the plot.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list with 7 elements:</p><dl><dt>Final_data</dt>
<dd><p>Input data frame plus the weighted Delat Ct values (wDCt)</p></dd>

  <dt>lm_ANOVA</dt>
<dd><p>lm of factorial analysis-tyle</p></dd>

  <dt>lm_ANCOVA</dt>
<dd><p>lm of ANCOVA analysis-type</p></dd>

  <dt>ANOVA_table</dt>
<dd><p>ANOVA table</p></dd>

  <dt>ANCOVA_table</dt>
<dd><p>ANCOVA table</p></dd>

  <dt>FC Table</dt>
<dd><p>Table of FC values, significance and confidence interval and standard error with the lower and upper limits for the main factor levels.</p></dd>

  <dt>Bar plot of FC values</dt>
<dd><p>Bar plot of the fold change values for the main factor levels.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>Fold change (\(\Delta \Delta C_T\) method) analysis of qPCR data can be done using 
ANOVA (analysis of variance) and ANCOVA (analysis of covariance) by the 
<code>qpcrANOVAFC</code> function, for uni- or multi-factorial experiment data. 
If there are more than one factor, FC value calculations for 
the `mainFactor.column` and the statistical analysis is performed based on a full model factorial 
experiment by default. However, if `ancova` is defined for the `analysisType` argument,
FC values are calculated for the levels of the `mainFactor.column` and the other factors are 
used as covariate(s) in the analysis of variance, but we should consider ANCOVA table:
if the interaction between the main factor and covariate is significant, ANCOVA is not appropriate in this case. 
ANCOVA is basically used when a factor is affected by uncontrolled quantitative covariate(s). 
For example, suppose that wDCt of a target gene in a plant is affected by temperature. The gene may 
also be affected by drought. Since we already know that temperature affects the target gene, we are 
interested to know if the gene expression is also altered by the drought levels. We can design an 
experiment to understand the gene behavior at both temperature and drought levels at the same time. 
The drought is another factor (the covariate) that may affect the expression of our gene under the 
levels of the first factor i.e. temperature. The data of such an experiment can be analyzed by ANCOVA 
or using ANOVA based on a factorial experiment using <code>qpcrANOVAFC</code>. <code>qpcrANOVAFC</code> function performs FC 
analysis even there is only one factor (without covariate or factor  variable). Bar plot of fold changes 
(FC) values along with the standard errors are also returned by the <code>qpcrANOVAFC</code> function.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis of qPCR data
and the application of simple blocking in qPCR experiments. BMC bioinformatics 18, 1-11.</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85). doi:10.1186/1471-2105-7-85.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Ghader Mirzaghaderi</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">data_1factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, mainFactor.column <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span>, </span></span>
<span class="r-in"><span>fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#CDC673"</span>, <span class="st">"#EEDD82"</span><span class="op">)</span>, fontsizePvalue <span class="op">=</span> <span class="fl">5</span>, y.axis.adjust <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The L1 level was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SA 4.9393  2.4696     2     4  14.319 0.01502 *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SA 4.9393  2.4696     2     4  14.319 0.01502 *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       L1 1.0000 1.0000     0.0000 0.0000 0.0208   0.9857   1.0145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 L2 vs L1 2.1987 0.0285   * 0.9514 5.0812 0.4388   1.6221   2.9803</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 L3 vs L1 3.4661 0.0061  ** 1.4999 8.0101 0.0841   3.2698   3.6742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">data_2factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, mainFactor.column <span class="op">=</span> <span class="fl">2</span>, block <span class="op">=</span> <span class="cn">NULL</span>, </span></span>
<span class="r-in"><span>analysisType <span class="op">=</span> <span class="st">"ancova"</span>, fontsizePvalue <span class="op">=</span> <span class="fl">5</span>, y.axis.adjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The D0 level was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   Sum Sq Mean Sq NumDF  DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Drought          12.9530  6.4765     2 9.9998  41.691 1.408e-05 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genotype          3.0505  3.0505     1 9.9998  19.637  0.001272 ** </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Drought:Genotype  4.5454  2.2727     2 9.9998  14.630  0.001072 ** </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genotype  3.0505  3.0505     1    14   6.649 0.0218703 *  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Drought  12.9530  6.4765     2    14  14.117 0.0004398 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       D0 1.0000 1.0000     0.0000 0.0000 0.3445   0.7876   1.2697</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 D1 vs D0 1.0705 0.8051     0.5266 2.1763 0.2631   0.8920   1.2847</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 D2 vs D0 3.5967 0.0003 *** 1.7693 7.3116 0.3576   2.8071   4.6084</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Data from Lee et al., 2020, Here, the data set contains technical </span></span></span>
<span class="r-in"><span><span class="co"># replicates so we get mean of technical replicates first:</span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="meanTech.html">meanTech</a></span><span class="op">(</span><span class="va">Lee_etal2020qPCR</span>, groups <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">df</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, analysisType <span class="op">=</span> <span class="st">"ancova"</span>, block <span class="op">=</span> <span class="cn">NULL</span>, </span></span>
<span class="r-in"><span>mainFactor.column <span class="op">=</span> <span class="fl">2</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"skyblue"</span>, <span class="st">"#BFEFFF"</span><span class="op">)</span>, y.axis.adjust <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The D12 level was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DS         184.275  61.425     3    24 31.2576 1.862e-08 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor1     22.317  11.158     2    24  5.6783   0.00957 ** </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DS:factor1  25.186   4.198     6    24  2.1361   0.08615 .  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Sum Sq Mean Sq NumDF DenDF F value   Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor1  22.317  11.158     2    30  4.6269  0.01772 *  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DS      184.275  61.425     3    30 25.4703 2.17e-08 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        D12 1.0000 1.0000     0.0000 0.0000 0.6443   0.6398   1.5630</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 D15 vs D12 0.2148 0.0050  ** 0.0541 0.8536 0.3767   0.1654   0.2789</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 D18 vs D12 0.0231 0.0000 *** 0.0058 0.0919 0.6469   0.0148   0.0362</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  D7 vs D12 1.1232 0.8204     0.2826 4.4636 0.5824   0.7501   1.6818</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">data_2factorBlock</span>,  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, mainFactor.column <span class="op">=</span> <span class="fl">1</span>, </span></span>
<span class="r-in"><span>mainFactor.level.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S"</span>, <span class="st">"R"</span><span class="op">)</span>, block <span class="op">=</span> <span class="st">"block"</span>, </span></span>
<span class="r-in"><span>fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#CDC673"</span>, <span class="st">"#EEDD82"</span><span class="op">)</span>, analysisType <span class="op">=</span> <span class="st">"ancova"</span>,</span></span>
<span class="r-in"><span>fontsizePvalue <span class="op">=</span> <span class="fl">7</span>, y.axis.adjust <span class="op">=</span> <span class="fl">0.1</span>, width <span class="op">=</span> <span class="fl">0.35</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> boundary (singular) fit: see help('isSingular')</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> block            0.0035  0.0035     1     1  0.0228  0.904592    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor1          3.0505  3.0505     1    10 19.6368  0.001271 ** </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor2         12.9530  6.4765     2    10 41.6916 1.408e-05 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor1:factor2  4.5454  2.2727     2    10 14.6303  0.001072 ** </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> block    0.0072  0.0072     1    13  0.0146 0.9055452    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor2 12.9530  6.4765     2    13 13.1231 0.0007602 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> factor1  3.0505  3.0505     1    13  6.1810 0.0272912 *  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        S 1.0000 1.0000     0.0000 0.0000 0.4288   0.7429   1.3461</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   R vs S 0.5651 0.0273   * 0.3442 0.9279 0.2920   0.4616   0.6919</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="meanTech.html">meanTech</a></span><span class="op">(</span><span class="va">Lee_etal2020qPCR</span>, groups <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span> </span></span>
<span class="r-in"><span><span class="va">df2</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">factor1</span> <span class="op">==</span> <span class="st">"DSWHi"</span>,<span class="op">]</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">df2</span>, mainFactor.column <span class="op">=</span> <span class="fl">1</span>, fontsizePvalue <span class="op">=</span> <span class="fl">5</span>, y.axis.adjust <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>block <span class="op">=</span> <span class="cn">NULL</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, analysisType <span class="op">=</span> <span class="st">"anova"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The D12 level was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DS 63.493  21.164     3     6  25.216 0.0008438 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DS 63.493  21.164     3     6  25.216 0.0008438 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        D12 1.0000 1.0000     0.0000  0.0000 0.6704   0.6283   1.5915</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 D15 vs D12 0.3139 0.0668   . 0.0522  1.8893 0.0888   0.2952   0.3338</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 D18 vs D12 0.0403 0.0008 *** 0.0067  0.2425 0.8773   0.0219   0.0740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  D7 vs D12 2.9851 0.0795   . 0.4960 17.9669 0.8740   1.6288   5.4709</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">addline_format</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">...</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'\\s'</span>,<span class="st">'\n'</span>,<span class="va">x</span><span class="op">)</span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="fu">qpcrANOVAFC</span><span class="op">(</span><span class="va">data_1factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, mainFactor.column <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>block <span class="op">=</span> <span class="cn">NULL</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"skyblue"</span>,<span class="st">"#79CDCD"</span><span class="op">)</span>, y.axis.by <span class="op">=</span> <span class="fl">1</span>, </span></span>
<span class="r-in"><span>letter.position.adjust <span class="op">=</span> <span class="fl">0</span>, y.axis.adjust <span class="op">=</span> <span class="fl">1</span>, ylab <span class="op">=</span> <span class="st">"Fold Change"</span>, </span></span>
<span class="r-in"><span>fontsize <span class="op">=</span> <span class="fl">12</span>, x.axis.labels.rename <span class="op">=</span> <span class="fu">addline_format</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, </span></span>
<span class="r-in"><span>                                         <span class="st">"Treatment_1 vs Control"</span>, </span></span>
<span class="r-in"><span>                                         <span class="st">"Treatment_2 vs Control"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The L1 level was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANOVA table </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SA 4.9393  2.4696     2     4  14.319 0.01502 *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANCOVA table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SA 4.9393  2.4696     2     4  14.319 0.01502 *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   contrast     FC pvalue sig    LCL    UCL     se Lower.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                  Control 1.0000 1.0000     0.0000 0.0000 0.0208   0.9857</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 Treatment_1\nvs\nControl 2.1987 0.0285   * 0.9514 5.0812 0.4388   1.6221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 Treatment_2\nvs\nControl 3.4661 0.0061  ** 1.4999 8.0101 0.0841   3.2698</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   1.0145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   2.9803</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   3.6742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change plot of the main factor levels</span>
<span class="r-plt img"><img src="qpcrANOVAFC-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span>                                                       </span></span>
<span class="r-in"><span>                                                       </span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Ghader Mirzaghaderi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

