<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Bar plot of the relative gene expression (\(\Delta C_T\) method) from the qpcrANOVARE output of a two-factorial experiment data — twoFACTORplot • rtpcr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Bar plot of the relative gene expression (\(\Delta C_T\) method) from the qpcrANOVARE output of a two-factorial experiment data — twoFACTORplot"><meta property="og:description" content="Bar plot of the relative expression (\(\Delta C_T\) method) of a gene along with the standard error (se), 95% confidence interval (ci) and significance"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rtpcr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/vignette.html">rtpcr: a package for statistical analysis of real-time PCR data in R</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mirzaghaderi/rtpcr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bar plot of the relative gene expression (\(\Delta C_T\) method) from the <code>qpcrANOVARE</code> output of a two-factorial experiment data</h1>
    <small class="dont-index">Source: <a href="https://github.com/mirzaghaderi/rtpcr/blob/HEAD/R/twoFACTORplot.r" class="external-link"><code>R/twoFACTORplot.r</code></a></small>
    <div class="hidden name"><code>twoFACTORplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Bar plot of the relative expression (\(\Delta C_T\) method) of a gene along with the standard error (se), 95% confidence interval (ci) and significance</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">twoFACTORplot</span><span class="op">(</span></span>
<span>  <span class="va">res</span>,</span>
<span>  <span class="va">x.axis.factor</span>,</span>
<span>  <span class="va">group.factor</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"Blues"</span>,</span>
<span>  y.axis.adjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  y.axis.by <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  show.errorbars <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">errorbar</span>,</span>
<span>  show.letters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  letter.position.adjust <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Relative Expression"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.09</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>  fontsize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  fontsizePvalue <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  axis.text.x.angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  axis.text.x.hjust <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>res</dt>
<dd><p>the FC data frame created by <code>qpcrANOVARE(x)$Result</code> function on a two factor data such as <code>data_2factor</code>.</p></dd>


<dt>x.axis.factor</dt>
<dd><p>x-axis factor.</p></dd>


<dt>group.factor</dt>
<dd><p>grouping factor.</p></dd>


<dt>width</dt>
<dd><p>a positive number determining bar width.</p></dd>


<dt>fill</dt>
<dd><p>specify the fill color vector for the columns of the bar plot. One of the palettes in <code><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">display.brewer.all</a></code> (e.g. "Reds" or "Blues", ...) can be applied.</p></dd>


<dt>y.axis.adjust</dt>
<dd><p>a negative or positive number for reducing or increasing the length of the y axis.</p></dd>


<dt>y.axis.by</dt>
<dd><p>determines y axis step length.</p></dd>


<dt>show.errorbars</dt>
<dd><p>show errorbars</p></dd>


<dt>errorbar</dt>
<dd><p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p></dd>


<dt>show.letters</dt>
<dd><p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p></dd>


<dt>letter.position.adjust</dt>
<dd><p>adjust the distance between the grouping letters to the error bars.</p></dd>


<dt>ylab</dt>
<dd><p>the title of the y axis.</p></dd>


<dt>xlab</dt>
<dd><p>the title of the x axis.</p></dd>


<dt>legend.position</dt>
<dd><p>a two digit vector specifying the legend position.</p></dd>


<dt>fontsize</dt>
<dd><p>size of all fonts  of the plot.</p></dd>


<dt>fontsizePvalue</dt>
<dd><p>font size of the pvalue labels</p></dd>


<dt>axis.text.x.angle</dt>
<dd><p>angle of x axis text</p></dd>


<dt>axis.text.x.hjust</dt>
<dd><p>horizontal justification of x axis text</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Bar plot of the average fold change for target genes along with the standard error or 95% confidence interval as error bars.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The <code>twoFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance, standard error (se) and the 95% confidence interval (ci) as error bars.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Ghader Mirzaghaderi</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># See a sample data frame</span></span></span>
<span class="r-in"><span><span class="va">data_2factor</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Genotype Drought Rep EPO  POCt EGAPDH GAPDHCt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         R      D0   1   2 33.30      2   31.53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2         R      D0   2   2 33.39      2   31.57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3         R      D0   3   2 33.34      2   31.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4         R      D1   1   2 32.73      2   31.30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5         R      D1   2   2 32.46      2   32.55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6         R      D1   3   2 32.60      2   31.92</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7         R      D2   1   2 33.48      2   33.30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8         R      D2   2   2 33.27      2   33.37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9         R      D2   3   2 33.32      2   33.35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10        S      D0   1   2 26.85      2   26.94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11        S      D0   2   2 28.17      2   27.69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12        S      D0   3   2 27.99      2   27.39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13        S      D1   1   2 30.41      2   28.70</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14        S      D1   2   2 29.49      2   28.66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15        S      D1   3   2 29.98      2   28.71</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16        S      D2   1   2 29.03      2   30.61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17        S      D2   2   2 28.73      2   30.20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18        S      D2   3   2 28.83      2   30.49</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Before generating plot, the result table needs to be extracted as below:</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="qpcrANOVARE.html">qpcrANOVARE</a></span><span class="op">(</span><span class="va">data_2factor</span>, numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">$</span><span class="va">Result</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis of Variance Table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Response: wDCt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T          5 20.5489  4.1098  26.267 4.515e-06 ***</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Residuals 12  1.8775  0.1565                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Relative expression table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Genotype Drought     RE    LCL    UCL     se Lower.se Upper.se letters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        S      D2 2.9690 2.1030 4.1918 0.0551   2.8577   3.0846    a   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2        R      D2 0.9885 0.7002 1.3956 0.0841   0.9325   1.0478     b  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3        S      D0 0.7955 0.5635 1.1232 0.2128   0.6864   0.9219     b  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4        R      D1 0.6271 0.4441 0.8853 0.4388   0.4626   0.8500     bc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5        S      D1 0.4147 0.2937 0.5854 0.2540   0.3478   0.4945      cd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6        R      D0 0.2852 0.2020 0.4026 0.0208   0.2811   0.2893       d</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Plot of the 'res' data with 'Genotype' as grouping factor</span></span></span>
<span class="r-in"><span><span class="fu">twoFACTORplot</span><span class="op">(</span><span class="va">res</span>,</span></span>
<span class="r-in"><span>   x.axis.factor <span class="op">=</span> <span class="va">Drought</span>,</span></span>
<span class="r-in"><span>   group.factor <span class="op">=</span> <span class="va">Genotype</span>,</span></span>
<span class="r-in"><span>   width <span class="op">=</span> <span class="fl">0.5</span>,</span></span>
<span class="r-in"><span>   fill <span class="op">=</span> <span class="st">"Greens"</span>,</span></span>
<span class="r-in"><span>   y.axis.adjust <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>   y.axis.by <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>   ylab <span class="op">=</span> <span class="st">"Relative Expression"</span>,</span></span>
<span class="r-in"><span>   xlab <span class="op">=</span> <span class="st">"Drought Levels"</span>,</span></span>
<span class="r-in"><span>   letter.position.adjust <span class="op">=</span> <span class="fl">0.2</span>,</span></span>
<span class="r-in"><span>   legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   errorbar <span class="op">=</span> <span class="st">"se"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="twoFACTORplot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span>   </span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Plotting the same data with 'Drought' as grouping factor</span></span></span>
<span class="r-in"><span><span class="fu">twoFACTORplot</span><span class="op">(</span><span class="va">res</span>,</span></span>
<span class="r-in"><span>              x.axis.factor <span class="op">=</span> <span class="va">Genotype</span>,</span></span>
<span class="r-in"><span>              group.factor <span class="op">=</span> <span class="va">Drought</span>,</span></span>
<span class="r-in"><span>              xlab <span class="op">=</span> <span class="st">"Genotype"</span>,</span></span>
<span class="r-in"><span>              fill <span class="op">=</span> <span class="st">"Blues"</span>,</span></span>
<span class="r-in"><span>              fontsizePvalue <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>              errorbar <span class="op">=</span> <span class="st">"ci"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="twoFACTORplot-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span>              </span></span>
<span class="r-in"><span>              </span></span>
<span class="r-in"><span>              </span></span>
<span class="r-in"><span><span class="co"># Combining FC results of two different genes:</span></span></span>
<span class="r-in"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_1</span>,</span></span>
<span class="r-in"><span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                  factor <span class="op">=</span> <span class="st">"time"</span>, block <span class="op">=</span> <span class="cn">NULL</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The level 1  of the selected factor was used as calibrator.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> time 11.073  5.5364     2     4  4.5382 0.09357 .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="qpcrREPEATED.html">qpcrREPEATED</a></span><span class="op">(</span><span class="va">data_repeated_measure_2</span>,</span></span>
<span class="r-in"><span>                  factor <span class="op">=</span> <span class="st">"time"</span>,</span></span>
<span class="r-in"><span>                  numberOfrefGenes <span class="op">=</span> <span class="fl">1</span>, block <span class="op">=</span> <span class="cn">NULL</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The level 1  of the selected factor was used as calibrator.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> NOTE: Results may be misleading due to involvement in interactions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Type III Analysis of Variance Table with Satterthwaite's method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> time           5.9166 2.95832     2     8  3.4888 0.08139 .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> treatment      0.6773 0.67728     1     4  0.7987 0.42199  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> time:treatment 6.3186 3.15932     2     8  3.7258 0.07186 .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fold Change table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         contrast     FC pvalue sig    LCL    UCL     se Lower.se Upper.se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          time1 1.0000 1.0000     0.0000 0.0000 0.7036   0.6140   1.6286</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 time2 vs time1 1.2556 0.5540     0.4381 3.5987 0.5323   0.8682   1.8159</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 time3 vs time1 2.5432 0.0351   * 0.8873 7.2895 0.7074   1.5575   4.1527</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">a1</span> <span class="op">&lt;-</span> <span class="va">a</span><span class="op">$</span><span class="va">FC_statistics_of_the_main_factor</span></span></span>
<span class="r-in"><span><span class="va">b1</span> <span class="op">&lt;-</span> <span class="va">b</span><span class="op">$</span><span class="va">FC_statistics_of_the_main_factor</span></span></span>
<span class="r-in"><span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">a1</span>, <span class="va">b1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">c</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">c</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         contrast     FC pvalue sig    LCL     UCL     se Lower.se Upper.se gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          time1 1.0000 1.0000     0.0000  0.0000 1.4051   0.3776   2.6484    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 time2 vs time1 0.8566 0.8166     0.0923  7.9492 0.9753   0.4357   1.6841    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 time3 vs time1 4.7022 0.0685   . 0.5067 43.6368 0.5541   3.2026   6.9040    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4          time1 1.0000 1.0000     0.0000  0.0000 0.7036   0.6140   1.6286    2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 time2 vs time1 1.2556 0.5540     0.4381  3.5987 0.5323   0.8682   1.8159    2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 time3 vs time1 2.5432 0.0351   * 0.8873  7.2895 0.7074   1.5575   4.1527    2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">twoFACTORplot</span><span class="op">(</span><span class="va">c</span>, x.axis.factor <span class="op">=</span> <span class="va">contrast</span>, </span></span>
<span class="r-in"><span>              group.factor <span class="op">=</span> <span class="va">gene</span>, fill <span class="op">=</span> <span class="st">'Reds'</span>,</span></span>
<span class="r-in"><span>              ylab <span class="op">=</span> <span class="st">"FC"</span>, axis.text.x.angle <span class="op">=</span> <span class="fl">45</span>,</span></span>
<span class="r-in"><span>              errorbar <span class="op">=</span> <span class="st">"se"</span>, y.axis.adjust <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>              axis.text.x.hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="twoFACTORplot-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Ghader Mirzaghaderi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

